<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • MAGMA.Celltyping</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.3" id="magmacelltyping-203">MAGMA.Celltyping 2.0.3<a class="anchor" aria-label="anchor" href="#magmacelltyping-203"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-2-0-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-2-0-3"></a></h3>
<ul><li>Fix GHA release/devel conflicts.</li>
<li>Add optional error handling with GitHub PAT fix instructions to
<code>get_data</code>, specifically <code>get_genomeLocFile</code>.</li>
<li>Construct URLs with <code>paste</code> instead of
<code>file.path</code> to avoid cross-platforms issues (e.g. Windows
inserting “" instead of”/“).</li>
<li>Add <code>dir.create</code> to <code>get_data</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.2" id="magmacelltyping-202">MAGMA.Celltyping 2.0.2<a class="anchor" aria-label="anchor" href="#magmacelltyping-202"></a></h2>
<div class="section level3">
<h3 id="new-features-2-0-2">New features<a class="anchor" aria-label="anchor" href="#new-features-2-0-2"></a></h3>
<ul><li>
<code>get_genomeLocFile</code>: Store NCBI ref file in cache dir by
default.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-2-0-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-2-0-2"></a></h3>
<ul><li>Add windows-specific files/folders to <em>.Rbuildignore</em>.<br></li>
<li>Bump <em>Depends</em> to R(&gt;= 4.1).</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.1" id="magmacelltyping-201">MAGMA.Celltyping 2.0.1<a class="anchor" aria-label="anchor" href="#magmacelltyping-201"></a></h2>
<div class="section level3">
<h3 id="new-features-2-0-1">New features<a class="anchor" aria-label="anchor" href="#new-features-2-0-1"></a></h3>
<ul><li>
<code>get_example_gwas_raw</code>: Allow users to specify
<code>timeout</code> when downloading raw (pre-munged) files.</li>
<li>Added <a href="https://neurogenomics.github.io/MAGMA_Celltyping/articles/docker" class="external-link">Docker/Singularity</a>
vignette, with link in README.<br></li>
<li>Add <em>–as-cran</em> to R CMD check in GHA.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-2-0-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-2-0-1"></a></h3>
<ul><li>Add <code>$annot</code> data.frames to each ctd in
<code>get_ctd</code> to make compatible with all <code>EWCE</code>
functions.<br></li>
<li>Add “ctd_DescartesHuman” to documented datasets in
<code>get_ctd</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.0" id="magmacelltyping-200">MAGMA.Celltyping 2.0.0<a class="anchor" aria-label="anchor" href="#magmacelltyping-200"></a></h2>
<ul><li>MAJOR UPGRADE: <code>MAGMA.Celltyping</code> was revamped to meet
CRAN standards, automatically install MAGMA, and take any species are
input.</li>
</ul><div class="section level3">
<h3 id="new-features-2-0-0">New features<a class="anchor" aria-label="anchor" href="#new-features-2-0-0"></a></h3>
<ul><li>Added a <code>NEWS.md</code> file to track changes to the
package.</li>
<li>Automatically install MAGMA with new <code>magma_install</code>
function; stores binaries in <code>MAGMA.Celltyping</code>-specific
cache dir. Added various support functions to make this possible and
ensure correct version is being used.</li>
<li>Added <code>magma_uninstall</code> function to remove one or all
MAGMA binaries.</li>
<li>Allow <code>MAGMA.Celltyping</code> to install even if MAGMA is not
installed. Instead, check at the beginning of functions that require
MAGMA using <code>magma_check</code>.
<ul><li>
<code>magma_links_stored</code>: Include built-in metadata with
links to all MAGMA versions with parsed version numbers, OS, and which
is the latest version.</li>
</ul></li>
<li>Call MAGMA commands with <code>magma_run</code> which finds the
requested version of MAGMA and uses it.</li>
<li>Print readable, user-friendly MAGMA commands when being run through
<code>magma_cmd</code> function.</li>
<li>Added unit tests.</li>
<li>Create hex sticker</li>
<li>New function: <code>get_sub_SNP_LOC_DATA</code>
</li>
<li>Formally deprecated functions using <code>.Deprecated</code>
function and removing all other internal code:
<ul><li><code>get_genomebuild_for_sumstats</code></li>
<li><code>build_snp_location_tables</code></li>
<li><code>format.sumstats.for.magma</code></li>
<li><code>format_sumstats_for_magma_macOnly</code></li>
<li><code>standardise.sumstats.column.headers</code></li>
<li><code>standardise.sumstats.column.headers.crossplatform</code></li>
</ul></li>
<li>Removed <code>sumstatsColHeaders</code> from data, as it was only
used in now-deprecated functions.<br></li>
<li>Renamed all functions with “.” to “_” to meet coding standards.</li>
<li>Renamed functions to be more concise and avoid issues with test file
names being too long:
<ul><li>
<code>calculate.celltype.enrichment.probabilities.wtLimma</code>
–&gt; <code>calculate_celltype_enrichment_limma</code>
</li>
<li>
<code>calculate.conditional.celltype.enrichment.probabilities.wtLimma</code>
–&gt; <code>calculate_conditional_celltype_enrichment_limma</code>
</li>
</ul></li>
<li>Removed all large <em>data/</em> to GitHub Releases, now accessible
with dedicated <code>piggyback</code>-based functions:
<ul><li>
<code>get_ctd</code>: CellTypeDatasets</li>
<li>
<code>get_example_gwas</code>: GWAS summary stats</li>
<li>
<code>get_genomeLocFile</code>: NCBI gene coordinate
references.</li>
</ul></li>
<li>Create example full GWAS summary stats (both unfiltered and filtered
+ munged with <code>MungeSumstats</code>). Accessed by
<code>get_example_gwas</code>.
<ul><li>“prospective_memory”</li>
<li>“fluid_intelligence”</li>
<li>“educational_attainment”</li>
</ul></li>
<li>Updated vignettes:
<ul><li>Created concise <em>Getting started</em> vignette.</li>
<li>Updated origiinal vignette and turned into <em>full_workflow</em>
vignette.</li>
</ul></li>
<li>Made certain functions run automatically internally, instead of
having the user run them:
<ul><li><code>get_genome_ref</code></li>
<li><code>prepare_quantile_groups</code></li>
</ul></li>
<li>Remove unnecessary dependencies:
<ul><li><code>reshape</code></li>
<li><code>cowplot</code></li>
<li><code>SNPlocs.Hsapiens.dbSNP144.GRCh37</code></li>
<li><code>SNPlocs.Hsapiens.dbSNP144.GRCh38</code></li>
</ul></li>
<li>Replaced <code>hgnc2entrez</code> with improved
<code>hgnc2entrez_ortohgene</code> from
<code><a href="https://rdrr.io/pkg/orthogene/man/all_genes.html" class="external-link">orthogene::all_genes</a></code>. Benchmarked to confirm that the
latter increases the number of genes that can be converted.</li>
<li>Allow all functions to accept datasets/gene lists from any species.
Now automatically converted to <code>output_species</code> (default:
“human”) using <a href="https://github.com/neurogenomics/orthogene" class="external-link"><code>orthogene</code></a>.</li>
<li>Create MAGMA files repository using various OpenGWAS datasets that
have been munged with <code>MungeSumstats</code>: <a href="https://github.com/neurogenomics/MAGMA_Files_Public" class="external-link">MAGMA_Files_Public</a>
<ul><li>
<code>magma_files_metadata</code>: Built-in table of all
pre-processed MAGMA files currently in the database.<br></li>
</ul></li>
<li>Added API to search and access MAGMA files repository:
<code>import_magma_files</code>.</li>
<li>Allow all relevant functions to take only MAGMA files as input
(instead of requiring the GWAS summary stats);
e.g. <code>calculate_celltype_associations(magma_dir="&lt;folder_containing_magma_files&gt;")</code>
This function is also used for downloading MAGMA files in examples/unit
tests.</li>
<li>Add header notation in code comments to improve code
navigability.</li>
<li>Fix Roxygen notes:
<ul><li>Document
<code>@title</code>,<code>@description</code>,<code>@param</code>,
<code>@return</code> for all exported (and many internal)
functions.</li>
<li>Document <code>@examples</code> for all exported (and many internal)
functions.</li>
<li>Used <code>@importFrom</code> or <code>requireNamespace</code> for
all imports functions.</li>
</ul></li>
<li>Replace usage of all <code>1:10</code> syntax.</li>
<li>Reduce number of functions in NAMESPACE</li>
<li>Set all defaults consistently across all functions:
<ul><li><code>upstream_kb = 35</code></li>
<li><code>downstream_kb = 10</code></li>
</ul></li>
<li>Allow the use of non-European populations by downloading
population-specific LD panels from 1KG with
<code>get_genome_ref(population = "&lt;population_name&gt;")</code>
</li>
<li>Handle other CTD matrix input types by ensuring standardisation as
dense matrices when computing quantiles/normalization.</li>
<li>Take advantage of new <code>EWCE</code> features in
<em>bschilder_dev</em> branch:
<ul><li>Standardise CTD internally in all relevant functions using new
<code><a href="https://rdrr.io/pkg/EWCE/man/standardise_ctd.html" class="external-link">EWCE::standardise_ctd</a></code>
</li>
</ul></li>
<li>Create all-in-one functions
<code>celltype_associations_pipeline</code>, which lets users specify
which test they want to run with arguments, including:
<ul><li>
<code>calculate_celltype_associations</code> (Linear mode)</li>
<li>
<code>calculate_celltype_associations</code> (Top10% mode)</li>
<li><code>calculate_conditional_celltype_associations</code></li>
</ul></li>
<li>Parallelise <code>celltype_associations_pipeline</code> across
multiple cores.</li>
<li>Removed old functions whose output were not being used:
<ul><li><code>normalise_mean_exp</code></li>
<li><code>bin_specificityDistance_into_quantiles</code></li>
<li><code>bin_expression_into_quantiles</code></li>
</ul></li>
<li>Add new function (plus tests):
<ul><li>
<code>get_driver_genes</code><br></li>
</ul></li>
<li>Added unit tests for:
<ul><li><code>calculate_celltype_enrichment_limma</code></li>
<li><code>adjust_zstat_in_genesOut</code></li>
<li>Deprecated functions</li>
</ul></li>
<li>Added R script to produce vignette results
<em>inst/extdata/MAGMA_Celltyping_1.0_vignette.R</em>, and uploaded
zipped folder via piggyback:
<em>MAGMA_Celltyping_1.0_results.zip</em><br></li>
<li>Added unit tests comparing old (1.0.0) vs new (&gt;=2.0)
MAGMA.Celltyping versions produce the same results;
<em>test-MAGMA_Celltyping_1.0_vs_2.0.R</em>. Full report <a href="https://github.com/neurogenomics/MAGMA_Celltyping/issues/96" class="external-link">here</a>.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-2-0-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-2-0-0"></a></h3>
<ul><li>Removed <code>usethis</code> call from code.</li>
<li>Removed all <code>library</code> calls from code.</li>
<li>Avoid accidentally renaming columns with
<code>data.frame</code>
</li>
<li>Remove all <code>suppressWarnings</code> calls and resolve the
underlying issues instead.</li>
<li>Add <code>utils</code> as Suggest.</li>
<li>Normalize paths to magma executables (to avoid path issues on
WindowsOS).</li>
<li>Fixed axes in <code>plot_celltype_associations</code>, first
reported <a href="https://github.com/neurogenomics/MAGMA_Celltyping/issues/12" class="external-link">here</a>.<br></li>
<li>Fixed <code>prepare_quantile_groups</code> so that it’s consistent
with how <code>EWCE</code> compute specificity quantiles. Ensures that
all celltypes (columns) have exactly the same number of quantiles, which
was not the case before.</li>
<li>Fixed bug in ``</li>
</ul></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

