<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Adjust MAGMA Z-statistic from .genes.out files — adjust_zstat_in_genesOut • MAGMA.Celltyping</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Adjust MAGMA Z-statistic from .genes.out files — adjust_zstat_in_genesOut"><meta property="og:description" content="Used when you want to directly analyse the gene-level
 Z-scores for a given GWAS while correcting 
 for known confounding variables such as:
NSNPS : Number of SNPs
NPARAM : Number of parameters?
GENELEN : Gene length
log*** : The logged version of each of the above variables, 
using the default log function.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Adjust MAGMA Z-statistic from <em>.genes.out</em> files</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/R/adjust_zstat_in_genesOut.r" class="external-link"><code>R/adjust_zstat_in_genesOut.r</code></a></small>
    <div class="hidden name"><code>adjust_zstat_in_genesOut.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Used when you want to directly analyse the gene-level
 Z-scores for a given GWAS while correcting 
 for known confounding variables such as:</p><ul><li><p><code>NSNPS</code> : Number of SNPs</p></li>
<li><p><code>NPARAM</code> : Number of parameters?</p></li>
<li><p><code>GENELEN</code> : Gene length</p></li>
<li><p><code>log***</code> : The logged version of each of the above variables, 
using the default <a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a> function.</p></li>
</ul></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">adjust_zstat_in_genesOut</span><span class="op">(</span></span>
<span>  <span class="va">magma_GenesOut_file</span>,</span>
<span>  ctd <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ctd_species <span class="op">=</span> <span class="fu"><a href="infer_ctd_species.html">infer_ctd_species</a></span><span class="op">(</span><span class="va">ctd</span><span class="op">)</span>,</span>
<span>  prepare_ctd <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>magma_GenesOut_file</dt>
<dd><p>A MAGMA <em>.genes.out</em> file 
generated by <a href="map_snps_to_genes.html">map_snps_to_genes</a>.</p></dd>


<dt>ctd</dt>
<dd><p>Cell type data structure containing
<code>specificity_quantiles</code>.</p></dd>


<dt>ctd_species</dt>
<dd><p>Species name relevant to the CellTypeDataset (<code>ctd</code>).
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species. 
If <code>ctd_species=NULL</code> (default),
 the <code>ctd</code> species will automatically 
 be inferred using <a href="https://rdrr.io/pkg/orthogene/man/infer_species.html" class="external-link">infer_species</a>.</p></dd>


<dt>prepare_ctd</dt>
<dd><p>Whether to run
<a href="prepare_quantile_groups.html">prepare_quantile_groups</a> on the <code>ctd</code> first.</p></dd>


<dt>method</dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/EWCE/man/standardise_ctd.html" class="external-link">EWCE::standardise_ctd</a></code></p><dl><dt><code>dataset</code></dt>
<dd><p>CellTypeData. name.</p></dd>

    <dt><code>input_species</code></dt>
<dd><p>Which species the gene names in <code>exp</code> come from.
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species.</p></dd>

    <dt><code>output_species</code></dt>
<dd><p>Which species' genes names to convert <code>exp</code> to.
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species.</p></dd>

    <dt><code>non121_strategy</code></dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>

    <dt><code>force_new_quantiles</code></dt>
<dd><p>By default, quantile computation is
skipped if they have already been computed.
Set <code>=TRUE</code> to override this and generate new quantiles.</p></dd>

    <dt><code>force_standardise</code></dt>
<dd><p>If <code>ctd</code> has already been standardised, whether
to rerun standardisation anyway (Default: <code>FALSE</code>).</p></dd>

    <dt><code>remove_unlabeled_clusters</code></dt>
<dd><p>Remove any samples that have
numeric column names.</p></dd>

    <dt><code>numberOfBins</code></dt>
<dd><p>Number of non-zero quantile bins.</p></dd>

    <dt><code>keep_annot</code></dt>
<dd><p>Keep the column annotation data if provided.</p></dd>

    <dt><code>keep_plots</code></dt>
<dd><p>Keep the dendrograms if provided.</p></dd>

    <dt><code>as_sparse</code></dt>
<dd><p>Convert to sparse matrix.</p></dd>

    <dt><code>as_DelayedArray</code></dt>
<dd><p>Convert to <code>DelayedArray</code>.</p></dd>

    <dt><code>rename_columns</code></dt>
<dd><p>Remove <code>replace_chars</code> from column names.</p></dd>

    <dt><code>make_columns_unique</code></dt>
<dd><p>Rename each columns with the prefix
<code>dataset.species.celltype</code>.</p></dd>

  
</dl></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">myGenesOut</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="import_magma_files.html">import_magma_files</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ieu-a-298"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    file_types <span class="op">=</span> <span class="st">".genes.out"</span>,</span></span>
<span class="r-in"><span>    return_dir <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using built-in example files: ieu-a-298.tsv.gz.35UP.10DOWN</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning MAGMA gene.* file paths</span>
<span class="r-in"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2023-02-28</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">magmaGenesOut</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu">adjust_zstat_in_genesOut</span><span class="op">(</span></span></span>
<span class="r-in"><span>    ctd <span class="op">=</span> <span class="va">ctd</span>,</span></span>
<span class="r-in"><span>    magma_GenesOut_file <span class="op">=</span> <span class="va">myGenesOut</span>,</span></span>
<span class="r-in"><span>    ctd_species <span class="op">=</span> <span class="st">"mouse"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 5 matrix types across 2 CTD levels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing genes.out file.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 genes without HGNC gene symbols were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 371 genes that are absent from the ctd were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing adjusted Z-statistic.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

