<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Use MAGMA to test enrichment in a geneset — calculate_conditional_geneset_enrichment • MAGMA.Celltyping</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Use MAGMA to test enrichment in a geneset — calculate_conditional_geneset_enrichment"><meta property="og:description" content='Run gene set enrichment analysis with a given geneset
on a GWAS previously mapped to genes
(using map_snps_to_genes),
while conditioning on a given cell-type or list of cell-types
(controlledCTs), as defined by the "specificity_quantiles" matrix
in a given level (controlledAnnotLevel) of the provide
CellTypeDataset (ctd). This allows one to conduct gene set enrichment
 analyses while controlling for strong cell-type-specific signatures.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
    <li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Use MAGMA to test enrichment in a geneset</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/master/R/calculate_conditional_geneset_enrichment.R" class="external-link"><code>R/calculate_conditional_geneset_enrichment.R</code></a></small>
    <div class="hidden name"><code>calculate_conditional_geneset_enrichment.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run gene set enrichment analysis with a given <code>geneset</code>
on a GWAS previously mapped to genes
(using <a href="map_snps_to_genes.html">map_snps_to_genes</a>),
while conditioning on a given cell-type or list of cell-types
(<code>controlledCTs</code>), as defined by the "specificity_quantiles" matrix
in a given level (<code>controlledAnnotLevel</code>) of the provide
CellTypeDataset (<code>ctd</code>). This allows one to conduct gene set enrichment
 analyses while controlling for strong cell-type-specific signatures.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">calculate_conditional_geneset_enrichment</span><span class="op">(</span></span>
<span>  <span class="va">geneset</span>,</span>
<span>  <span class="va">ctd</span>,</span>
<span>  ctd_species <span class="op">=</span> <span class="fu"><a href="infer_ctd_species.html">infer_ctd_species</a></span><span class="op">(</span><span class="va">ctd</span><span class="op">)</span>,</span>
<span>  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">controlledCTs</span>,</span>
<span>  prepare_ctd <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  gwas_sumstats_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  magma_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  analysis_name <span class="op">=</span> <span class="st">"MainRun"</span>,</span>
<span>  upstream_kb <span class="op">=</span> <span class="fl">35</span>,</span>
<span>  downstream_kb <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  geneset_species <span class="op">=</span> <span class="fu"><a href="infer_geneset_species.html">infer_geneset_species</a></span><span class="op">(</span><span class="va">geneset</span><span class="op">)</span>,</span>
<span>  version <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-geneset">geneset<a class="anchor" aria-label="anchor" href="#arg-geneset"></a></dt>
<dd><p>Genes which are to be tested.
Can be gene symbols (or other gene IDs) from any species listed in
<a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a>.</p></dd>


<dt id="arg-ctd">ctd<a class="anchor" aria-label="anchor" href="#arg-ctd"></a></dt>
<dd><p>Cell type data structure containing
<code>specificity_quantiles</code>.</p></dd>


<dt id="arg-ctd-species">ctd_species<a class="anchor" aria-label="anchor" href="#arg-ctd-species"></a></dt>
<dd><p>Species name relevant to the CellTypeDataset (<code>ctd</code>).
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species.
If <code>ctd_species=NULL</code> (default),
 the <code>ctd</code> species will automatically
 be inferred using <a href="https://rdrr.io/pkg/orthogene/man/infer_species.html" class="external-link">infer_species</a>.</p></dd>


<dt id="arg-controlledannotlevel">controlledAnnotLevel<a class="anchor" aria-label="anchor" href="#arg-controlledannotlevel"></a></dt>
<dd><p>Annotation level of the celltypes
being controlled for.</p></dd>


<dt id="arg-controlledcts">controlledCTs<a class="anchor" aria-label="anchor" href="#arg-controlledcts"></a></dt>
<dd><p>Array of the celltype to be controlled for,
e.g. c("Interneuron type 16","Medium Spiny Neuron).</p></dd>


<dt id="arg-prepare-ctd">prepare_ctd<a class="anchor" aria-label="anchor" href="#arg-prepare-ctd"></a></dt>
<dd><p>Whether to run
<a href="prepare_quantile_groups.html">prepare_quantile_groups</a> on the <code>ctd</code> first.</p></dd>


<dt id="arg-gwas-sumstats-path">gwas_sumstats_path<a class="anchor" aria-label="anchor" href="#arg-gwas-sumstats-path"></a></dt>
<dd><p>File path of the summary statistics file.</p></dd>


<dt id="arg-magma-dir">magma_dir<a class="anchor" aria-label="anchor" href="#arg-magma-dir"></a></dt>
<dd><p>Path to folder containing the
pre-computed MAGMA GWAS files (<em>.gsa.raw</em>and <em>.gsa.out</em>).</p></dd>


<dt id="arg-analysis-name">analysis_name<a class="anchor" aria-label="anchor" href="#arg-analysis-name"></a></dt>
<dd><p>Used in file names which area created.</p></dd>


<dt id="arg-upstream-kb">upstream_kb<a class="anchor" aria-label="anchor" href="#arg-upstream-kb"></a></dt>
<dd><p>How many kb upstream of the gene
should SNPs be included?</p></dd>


<dt id="arg-downstream-kb">downstream_kb<a class="anchor" aria-label="anchor" href="#arg-downstream-kb"></a></dt>
<dd><p>How many kb downstream of the gene
should SNPs be included?</p></dd>


<dt id="arg-geneset-species">geneset_species<a class="anchor" aria-label="anchor" href="#arg-geneset-species"></a></dt>
<dd><p>Species that the <code>geneset</code> came from.
Can be any species listed in <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a>.
If <code>geneset_species!="human"</code>, the <code>geneset</code> will be converted
 to 1:1 human orthologs using <a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">convert_orthologs</a>.</p></dd>


<dt id="arg-version">version<a class="anchor" aria-label="anchor" href="#arg-version"></a></dt>
<dd><p>MAGMA version to use.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a> of
 baseline (column name prefix "BASELINE_") and
 conditioned (column name prefix "COND_") gene set enrichment results.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Import data ####</span></span></span>
<span class="r-in"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_allKI"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: piggyback</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading 1 CellTypeDataset file(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-in"><span><span class="va">magma_dir</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="import_magma_files.html">import_magma_files</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="st">"ieu-a-298"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using built-in example files: ieu-a-298.tsv.gz.35UP.10DOWN</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning MAGMA directories.</span>
<span class="r-in"><span><span class="va">geneset</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="va"><a href="rbfox_binding.html">rbfox_binding</a></span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu">calculate_conditional_geneset_enrichment</span><span class="op">(</span></span></span>
<span class="r-in"><span>    geneset <span class="op">=</span> <span class="va">geneset</span>,</span></span>
<span class="r-in"><span>    ctd <span class="op">=</span> <span class="va">ctd</span>,</span></span>
<span class="r-in"><span>    controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    controlledCTs <span class="op">=</span> <span class="st">"pyramidal SS"</span>,</span></span>
<span class="r-in"><span>    magma_dir <span class="op">=</span> <span class="va">magma_dir</span>,</span></span>
<span class="r-in"><span>    analysis_name <span class="op">=</span> <span class="st">"Rbfox_16_pyrSS"</span>,  </span></span>
<span class="r-in"><span>    geneset_species <span class="op">=</span> <span class="st">"mouse"</span>, </span></span>
<span class="r-in"><span>    ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span> </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Installed MAGMA version: v1.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Skipping MAGMA installation.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The desired_version of MAGMA is currently installed: v1.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using: magma_v1.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd is already standardised. Returning original ctd.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Set force_standardise=TRUE to re-standardise.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> character format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to data.frame</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,143 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3,769 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3,769 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 13 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 2 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    398 / 4,143 (9.6%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    3,745 / 4,143 (90%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping gene symbols in specificity_quantiles matrix to entrez IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading sets out info from .gsa.out file.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

