<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculate celltype enrichments without MAGMA — calculate_conditional_celltype_enrichment_limma • MAGMA.Celltyping</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate celltype enrichments without MAGMA — calculate_conditional_celltype_enrichment_limma"><meta property="og:description" content="Calculate celltype enrichments without MAGMA
using adjusted MAGMA Z-statistic from .genes.out files."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate celltype enrichments without MAGMA</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/R/calculate_conditional_celltype_enrichment_limma.R" class="external-link"><code>R/calculate_conditional_celltype_enrichment_limma.R</code></a></small>
    <div class="hidden name"><code>calculate_conditional_celltype_enrichment_limma.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Calculate celltype enrichments without MAGMA
using adjusted MAGMA Z-statistic from .genes.out files.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">calculate_conditional_celltype_enrichment_limma</span><span class="op">(</span></span>
<span>  <span class="va">magma1</span>,</span>
<span>  <span class="va">magma2</span>,</span>
<span>  <span class="va">ctd</span>,</span>
<span>  ctd_species <span class="op">=</span> <span class="fu"><a href="infer_ctd_species.html">infer_ctd_species</a></span><span class="op">(</span><span class="va">ctd</span><span class="op">)</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  prepare_ctd <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  thresh <span class="op">=</span> <span class="fl">1e-04</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>magma1</dt>
<dd><p>Output from <a href="adjust_zstat_in_genesOut.html">adjust_zstat_in_genesOut</a>
(e.g. trait 1).</p></dd>


<dt>magma2</dt>
<dd><p>Output from <a href="adjust_zstat_in_genesOut.html">adjust_zstat_in_genesOut</a>
(e.g. trait 2).</p></dd>


<dt>ctd</dt>
<dd><p>Cell type data structure containing
<code>specificity_quantiles</code>.</p></dd>


<dt>ctd_species</dt>
<dd><p>Species name relevant to the CellTypeDataset (<code>ctd</code>).
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species. 
If <code>ctd_species=NULL</code> (default),
 the <code>ctd</code> species will automatically 
 be inferred using <a href="https://rdrr.io/pkg/orthogene/man/infer_species.html" class="external-link">infer_species</a>.</p></dd>


<dt>annotLevel</dt>
<dd><p>Level within the <code>ctd</code> to use.</p></dd>


<dt>prepare_ctd</dt>
<dd><p>Whether to run
<a href="prepare_quantile_groups.html">prepare_quantile_groups</a> on the <code>ctd</code> first.</p></dd>


<dt>thresh</dt>
<dd><p>A threshold on low specificity values (used to drop genes).</p></dd>


<dt>show_plot</dt>
<dd><p>Show plot visualizations of results (Default: <code>TRUE</code>).</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to <a href="https://rdrr.io/pkg/EWCE/man/standardise_ctd.html" class="external-link">standardise_ctd</a>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Data.frame with conditional enrichment results.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### MAGMA 1 #####</span></span></span>
<span class="r-in"><span><span class="va">genesOut1</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="import_magma_files.html">import_magma_files</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ieu-a-298"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    file_types <span class="op">=</span> <span class="st">".genes.out"</span>,</span></span>
<span class="r-in"><span>    return_dir <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using built-in example files: ieu-a-298.tsv.gz.35UP.10DOWN</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning MAGMA gene.* file paths</span>
<span class="r-in"><span><span class="va">magma1</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="adjust_zstat_in_genesOut.html">adjust_zstat_in_genesOut</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    ctd <span class="op">=</span> <span class="va">ctd</span>, </span></span>
<span class="r-in"><span>    magma_GenesOut_file <span class="op">=</span> <span class="va">genesOut1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd_species=NULL: Inferring species from gene names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dense matrix format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from rownames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 16,843 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 16,843 genes from monkey.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10116</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,616 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,616 genes from rat.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7955</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,897 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,897 genes from zebrafish.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 8,438 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 8,438 genes from fly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Top match:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - species: mouse </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - percent_match: 92%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Inferred ctd species: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 5 matrix types across 2 CTD levels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing genes.out file.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 genes without HGNC gene symbols were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 371 genes that are absent from the ctd were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing adjusted Z-statistic.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### MAGMA 2 #####</span></span></span>
<span class="r-in"><span><span class="va">genesOut2</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="import_magma_files.html">import_magma_files</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ukb-a-333"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    file_types <span class="op">=</span> <span class="st">".genes.out"</span>,</span></span>
<span class="r-in"><span>    return_dir <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering IDs to only 1 requested dataset(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading 1 files.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning MAGMA gene.* file paths</span>
<span class="r-in"><span><span class="va">magma2</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="adjust_zstat_in_genesOut.html">adjust_zstat_in_genesOut</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    ctd <span class="op">=</span> <span class="va">ctd</span>, </span></span>
<span class="r-in"><span>    magma_GenesOut_file <span class="op">=</span> <span class="va">genesOut2</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd_species=NULL: Inferring species from gene names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dense matrix format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from rownames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 16,843 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 16,843 genes from monkey.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10116</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,616 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,616 genes from rat.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7955</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,897 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,897 genes from zebrafish.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 8,438 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 8,438 genes from fly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Top match:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - species: mouse </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - percent_match: 92%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Inferred ctd species: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 5 matrix types across 2 CTD levels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing genes.out file.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 55 genes without HGNC gene symbols were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 5,799 genes that are absent from the ctd were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 genes with duplicated HGNC gene symbols were dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing adjusted Z-statistic.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### Run conditional enrichment ####</span></span></span>
<span class="r-in"><span><span class="va">cond_res</span> <span class="op">&lt;-</span> </span></span>
<span class="r-in"><span><span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu">calculate_conditional_celltype_enrichment_limma</span><span class="op">(</span></span></span>
<span class="r-in"><span>    magma1 <span class="op">=</span> <span class="va">magma1</span>,</span></span>
<span class="r-in"><span>    magma2 <span class="op">=</span> <span class="va">magma2</span>,</span></span>
<span class="r-in"><span>    ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Correlation between magma1 and magma2 ADJ_ZSTAT: 0.051</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Call:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> stats::lm(formula = ADJ_ZSTAT.x ~ ADJ_ZSTAT.y, data = m3)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Residuals:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Min      1Q  Median      3Q     Max </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -4.3930 -0.5094  0.0290  0.5759  3.1231 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Coefficients:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>             Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> (Intercept)  5.29185    0.02921 181.141   &lt;2e-16 ***</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ADJ_ZSTAT.y  0.03827    0.02433   1.573    0.116    </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ---</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Residual standard error: 0.9046 on 960 degrees of freedom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Multiple R-squared:  0.002571,	Adjusted R-squared:  0.001532 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> F-statistic: 2.474 on 1 and 960 DF,  p-value: 0.1161</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd_species=NULL: Inferring species from gene names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dense matrix format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from rownames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 16,843 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 16,843 genes from monkey.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10116</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,616 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,616 genes from rat.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7955</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,897 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,897 genes from zebrafish.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 8,438 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 8,438 genes from fly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Top match:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - species: mouse </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - percent_match: 92%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Inferred ctd species: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 5 matrix types across 2 CTD levels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing level: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: astrocytes_ependymal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: endothelial_mural</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: interneurons</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: microglia</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: oligodendrocytes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: pyramidal_CA1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running conditonal enrichment tests: pyramidal_SS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd is already standardised. Returning original ctd.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Set force_standardise=TRUE to re-standardise.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing specificity matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: astrocytes_ependymal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: endothelial_mural</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: interneurons</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: microglia</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: oligodendrocytes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: pyramidal_CA1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: pyramidal_SS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd is already standardised. Returning original ctd.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Set force_standardise=TRUE to re-standardise.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing specificity matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: astrocytes_ependymal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: endothelial_mural</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: interneurons</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: microglia</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: oligodendrocytes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: pyramidal_CA1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running enrichment tests: pyramidal_SS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-plt img"><img src="calculate_conditional_celltype_enrichment_limma-1.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="calculate_conditional_celltype_enrichment_limma-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

