<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Infer CellTypeDataset species — infer_ctd_species • MAGMA.Celltyping</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Infer CellTypeDataset species — infer_ctd_species"><meta property="og:description" content="Infers species from from level 1 of a CellTypeDataset (CTD)
using either the metadata stored in the CTD
 (if the object has previously been standardised with 
 standardise_ctd) or using the gene names
  (via infer_species). 
If ctd_species is not NULL, 
this will be returned instead of inferring the species."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Infer CellTypeDataset species</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/R/infer_ctd_species.R" class="external-link"><code>R/infer_ctd_species.R</code></a></small>
    <div class="hidden name"><code>infer_ctd_species.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Infers species from from level 1 of a CellTypeDataset (CTD)
using either the metadata stored in the CTD
 (if the object has previously been standardised with 
 <a href="https://rdrr.io/pkg/EWCE/man/standardise_ctd.html" class="external-link">standardise_ctd</a>) or using the gene names
  (via <a href="https://rdrr.io/pkg/orthogene/man/infer_species.html" class="external-link">infer_species</a>). 
If <code>ctd_species</code> is not <code>NULL</code>, 
this will be returned instead of inferring the species.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">infer_ctd_species</span><span class="op">(</span><span class="va">ctd</span>, ctd_species <span class="op">=</span> <span class="cn">NULL</span>, verbose <span class="op">=</span> <span class="fl">2</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ctd</dt>
<dd><p>Cell type data structure containing
<code>specificity_quantiles</code>.</p></dd>


<dt>ctd_species</dt>
<dd><p>Species name relevant to the CellTypeDataset (<code>ctd</code>).
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species. 
If <code>ctd_species=NULL</code> (default),
 the <code>ctd</code> species will automatically 
 be inferred using <a href="https://rdrr.io/pkg/orthogene/man/infer_species.html" class="external-link">infer_species</a>.</p></dd>


<dt>verbose</dt>
<dd><p>Message verbosity.</p><ul><li><p><code>0</code> or <code>FALSE</code> : 
Don't print any messages.</p></li>
<li><p><code>1</code> or <code>TRUE</code> : 
Only print messages from <span class="pkg">MAGMA.Celltyping</span>.</p></li>
<li><p><code>2</code> or <code>c(TRUE,TRUE)</code> : 
Print messages from <span class="pkg">MAGMA.Celltyping</span> and 
the internal <span class="pkg">orthogene</span> function.</p></li>
</ul></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/orthogene/man/infer_species.html" class="external-link">orthogene::infer_species</a></code></p><dl><dt><code>gene_df</code></dt>
<dd><p>Data object containing the genes
(see <code>gene_input</code> for options on how
the genes can be stored within the object).<br>
Can be one of the following formats:<br></p><ul><li><p><code>matrix</code> : <br>A sparse or dense matrix.</p></li>
<li><p><code>data.frame</code> : <br>A <code>data.frame</code>,
 <code>data.table</code>. or <code>tibble</code>.</p></li>
<li><p>codelist : <br>A <code>list</code> or character <code>vector</code>.</p></li>
</ul><p>Genes, transcripts, proteins, SNPs, or genomic ranges
 can be provided in any format
(HGNC, Ensembl, RefSeq, UniProt, etc.) and will be
automatically converted to gene symbols unless
specified otherwise with the <code>...</code> arguments.<br><em>Note</em>: If you set <code>method="homologene"</code>, you
must either supply genes in gene symbol format (e.g. "Sox2")
 OR set <code>standardise_genes=TRUE</code>.</p></dd>

    <dt><code>gene_input</code></dt>
<dd><p>Which aspect of <code>gene_df</code> to
get gene names from:<br></p><ul><li><p><code>"rownames"</code> : <br>From row names of data.frame/matrix.</p></li>
<li><p><code>"colnames"</code> : <br>From column names of data.frame/matrix.</p></li>
<li><p><code>&lt;column name&gt;</code> : <br>From a column in <code>gene_df</code>,
 e.g. <code>"gene_names"</code>.</p></li>
</ul></dd>

    <dt><code>test_species</code></dt>
<dd><p>Which species to test for matches with.
If set to <code>NULL</code>, will default to a list of humans and
 5 common model organisms.
 If <code>test_species</code> is set to one of the following options, 
 it will automatically pull all species from that respective package and 
 test against each of them:</p><ul><li><p>"homologene" : 20+ species (default)</p></li>
<li><p>"gprofiler" : 700+ species</p></li>
<li><p>"babelgene" : 19 species</p></li>
</ul></dd>

    <dt><code>method</code></dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>

    <dt><code>make_plot</code></dt>
<dd><p>Make a plot of the results.</p></dd>

    <dt><code>show_plot</code></dt>
<dd><p>Print the plot of the results.</p></dd>

  
</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Inferred species name.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span>  </span></span>
<span class="r-in"><span><span class="va">ctd_species</span> <span class="op">&lt;-</span> <span class="fu">infer_ctd_species</span><span class="op">(</span>ctd <span class="op">=</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ctd_species=NULL: Inferring species from gene names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dense matrix format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from rownames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 16,843 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 16,843 genes from monkey.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for rat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10116</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,616 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,616 genes from rat.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for zebrafish</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7955</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,897 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,897 genes from zebrafish.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for gene overlap with: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 8,438 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 8,438 genes from fly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Top match:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - species: mouse </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - percent_match: 92%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Inferred ctd species: mouse</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

