<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Map SNPs to their nearby genes — map_snps_to_genes • MAGMA.Celltyping</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Map SNPs to their nearby genes — map_snps_to_genes"><meta property="og:description" content="Make two external calls to MAGMA. First use it to annotate SNPs
 onto their neighbouring genes. Second, use it to calculate
 the gene level trait association."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
    <li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Map SNPs to their nearby genes</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/master/R/map_snps_to_genes.r" class="external-link"><code>R/map_snps_to_genes.r</code></a></small>
    <div class="hidden name"><code>map_snps_to_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Make two external calls to MAGMA. First use it to annotate SNPs
 onto their neighbouring genes. Second, use it to calculate
 the gene level trait association.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">map_snps_to_genes</span><span class="op">(</span></span>
<span>  <span class="va">path_formatted</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  upstream_kb <span class="op">=</span> <span class="fl">35</span>,</span>
<span>  downstream_kb <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  N <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  duplicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"drop"</span>, <span class="st">"first"</span>, <span class="st">"last"</span>, <span class="st">"error"</span><span class="op">)</span>,</span>
<span>  synonym_dup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"skip"</span>, <span class="st">"skip-dup"</span>, <span class="st">"drop"</span>, <span class="st">"drop-dup"</span>, <span class="st">"error"</span><span class="op">)</span>,</span>
<span>  genome_ref_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  population <span class="op">=</span> <span class="st">"eur"</span>,</span>
<span>  genes_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  storage_dir <span class="op">=</span> <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/userdir.html" class="external-link">R_user_dir</a></span><span class="op">(</span><span class="st">"MAGMA.Celltyping"</span>, which <span class="op">=</span> <span class="st">"cache"</span><span class="op">)</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  version <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-path-formatted">path_formatted<a class="anchor" aria-label="anchor" href="#arg-path-formatted"></a></dt>
<dd><p>Filepath of the summary statistics file
(which is expected to already be in the required format).
Can be uncompressed or compressed (".gz" or ".bgz").</p></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>The build of the reference genome
 (<code>"GRCh37"</code> or <code>"GRCh38"</code>).
If <code>NULL</code>, it will be inferred with
<a href="https://al-murphy.github.io/MungeSumstats/reference/get_genome_build.html" class="external-link">get_genome_build</a>.</p></dd>


<dt id="arg-upstream-kb">upstream_kb<a class="anchor" aria-label="anchor" href="#arg-upstream-kb"></a></dt>
<dd><p>How many kilobases (kb) upstream of the gene
should SNPs be included?</p></dd>


<dt id="arg-downstream-kb">downstream_kb<a class="anchor" aria-label="anchor" href="#arg-downstream-kb"></a></dt>
<dd><p>How many kilobases (kb) downstream of the gene
should SNPs be included?</p></dd>


<dt id="arg-n">N<a class="anchor" aria-label="anchor" href="#arg-n"></a></dt>
<dd><p>What is the N number for this GWAS? That is cases + controls.</p></dd>


<dt id="arg-duplicate">duplicate<a class="anchor" aria-label="anchor" href="#arg-duplicate"></a></dt>
<dd><p>The duplicate modifier can be used to specify the desired
behaviour for dealing with duplicate SNPs in the file, and can be set to
one of four values: 'drop', 'first', 'last', and 'error'. When set to
'drop', the corresponding SNP is removed from the analysis entirely.
When set to 'first' or 'last', either the first or the last entry for
 that SNPs in the file is used. When set to 'error', the program terminates
 if encountering any duplicate SNPs. The default mode is 'duplicate=drop'.
 Note that SNPs are only checked for duplication if they are present in
 the genotype data, and if they have a non-missing pvalue
 (and sample size, if ncol is set). When synonymous SNP IDs have been
 loaded, different SNP IDs referring to the same SNP are considered
 duplicates as well. Unless duplicate is set to 'error', a list of
 duplicate SNPs will be written to the supplementary log file.</p></dd>


<dt id="arg-synonym-dup">synonym_dup<a class="anchor" aria-label="anchor" href="#arg-synonym-dup"></a></dt>
<dd><p>When loading SNP ID synonyms, MAGMA may detect SNP IDs
in the genotype data that are synonyms of each other. The synonym-dup
 modifier for the –bfile flag can be used to specify the desired behaviour
 for dealing with such SNPs. This modifier can be set to one of four values:
  'drop', 'drop-dup', 'skip', 'skip-dup' and 'error'. When set to 'drop',
  SNPs that have multiple synonyms in the data are removed from the analysis.
   Conversely, when set to 'skip' the SNPs are left in the data and the
   synonym entry in the synonym file is skipped. When set to 'drop-dup',
   for each synonym entry only the first listed in the synonym file is
   retained; for subsequent SNP IDs in the same entry that are found in
   the data are removed, and their IDs are mapped as synonyms to the first
    SNP. When set to 'skipdup' the genotype data for all synonymous SNPs
    is retained; SNP IDs not found in the data are mapped to the first SNP
     in the synonym entry that is. Finally, when set to 'error', the program
      will simply terminate when encountering synonymous SNPs in the data.
      The default mode is 'synonym-dup=skip'. Unless synonym-dup is set to
      error, a list of synonymous SNPs in the data will be written to the
      supplementary log file.</p></dd>


<dt id="arg-genome-ref-path">genome_ref_path<a class="anchor" aria-label="anchor" href="#arg-genome-ref-path"></a></dt>
<dd><p>Path to the folder containing the 1000
genomes reference (downloaded with <a href="get_genome_ref.html">get_genome_ref</a>).</p></dd>


<dt id="arg-population">population<a class="anchor" aria-label="anchor" href="#arg-population"></a></dt>
<dd><p>Which population subset of the genome reference
to include.</p><ul><li><p>"eur" : European descent
(Default simply because this is currently
the most common GWAS subpopulation).</p></li>
<li><p>"afr" : African descent.</p></li>
<li><p>"amr" : Ad Mixed American descent.</p></li>
<li><p>"eas" : East Asian descent.</p></li>
<li><p>"sas" : South Asian descent.</p></li>
</ul></dd>


<dt id="arg-genes-only">genes_only<a class="anchor" aria-label="anchor" href="#arg-genes-only"></a></dt>
<dd><p>The <em>.genes.raw</em> file is the intermediary file
that serves as  the input for subsequent gene-level analyses.
 To perform only a gene analysis, with no subsequent gene-set analysis,
 the <code>--genes-only</code> flag can be added (<code>TRUE</code>).
 This suppresses the creation of the <em>.genes.raw</em> file,
 and significantly reduces the running time and memory required.</p></dd>


<dt id="arg-storage-dir">storage_dir<a class="anchor" aria-label="anchor" href="#arg-storage-dir"></a></dt>
<dd><p>Where to store genome ref.</p></dd>


<dt id="arg-force-new">force_new<a class="anchor" aria-label="anchor" href="#arg-force-new"></a></dt>
<dd><p>Set to <code>TRUE</code> to
rerun <code>MAGMA</code> even if the output files already exist.
 (Default: <code>FALSE</code>).</p></dd>


<dt id="arg-version">version<a class="anchor" aria-label="anchor" href="#arg-version"></a></dt>
<dd><p>MAGMA version to use.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Path to the genes.out file.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="va">path_formatted</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="get_example_gwas.html">get_example_gwas</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">genesOutPath</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu">map_snps_to_genes</span><span class="op">(</span></span></span>
<span class="r-in"><span>    path_formatted <span class="op">=</span> <span class="va">path_formatted</span>,</span></span>
<span class="r-in"><span>    genome_build <span class="op">=</span> <span class="st">"hg19"</span>,</span></span>
<span class="r-in"><span>    N <span class="op">=</span> <span class="fl">5000</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># } </span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

