<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MAGMA_Celltyping: full_workflow • MAGMA.Celltyping</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MAGMA_Celltyping: full_workflow">
<meta property="og:description" content="MAGMA.Celltyping">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MAGMA.Celltyping.html">MAGMA_Celltyping: Getting started</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">MAGMA_Celltyping: full_workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MAGMA_Celltyping: full_workflow</h1>
                        <h4 data-toc-skip class="author"><h4>
Authors: <i>Brian M. Schilder, Alan Murphy, Julien Bryois &amp; Nathan Skene</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: <i>Feb-01-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/vignettes/full_workflow.Rmd" class="external-link"><code>vignettes/full_workflow.Rmd</code></a></small>
      <div class="hidden name"><code>full_workflow.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/MAGMA_Celltyping" class="external-link">MAGMA.Celltyping</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ===== Welcome to MAGMA.Celltyping =====</span>
<span class="co">## This package depends on the MAGMA command line tool:</span>
<span class="co">## https://ctg.cncr.nl/software/magma</span>
<span class="co">## ===== =========================== =====</span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'dplyr'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     filter, lag</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     intersect, setdiff, setequal, union</span></code></pre>
<div class="section level2">
<h2 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p><code>MAGMA.Celltyping</code> is a software package that facilitates conducting cell-type-specific enrichment tests on GWAS summary statistics.</p>
<p>The following vignette shows the step-by-step version of the <code>MAGMA.Celltyping</code> pipeline. For the newer streamlined version of the pipeline (that wraps many of these functions), please see the <em>Getting Started</em> vignette.</p>
</div>
<div class="section level2">
<h2 id="prepare-data">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h2>
<div class="section level3">
<h3 id="import-gwas">Import GWAS<a class="anchor" aria-label="anchor" href="#import-gwas"></a>
</h3>
<ul>
<li>We need to have a GWAS summary statistics file to analyse as input.<br>
</li>
<li>As an example, you can download UK Biobank summary statistics for ‘fluid_intelligence’ using <code><a href="../reference/get_example_gwas.html">get_example_gwas()</a></code>.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwas_sumstats_path</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_example_gwas.html">get_example_gwas</a></span><span class="op">(</span>
  trait <span class="op">=</span> <span class="st">"fluid_intelligence"</span>,
  munged <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="munge-gwas">Munge GWAS<a class="anchor" aria-label="anchor" href="#munge-gwas"></a>
</h3>
<p>Our lab have created a robust Bioconductor package for formatting multiple types of summary statistics files: <a href="https://github.com/neurogenomics/MungeSumstats" class="external-link"><code>MungeSumstats</code></a> (please cite the associated publication if you use this package):</p>
<blockquote>
<p><a href="https://doi.org/10.1093/bioinformatics/btab665" class="external-link">Murphy, Schilder, &amp; Skene, MungeSumstats: a Bioconductor package for the standardization and quality control of many GWAS summary statistics, Bioinformatics, Volume 37, Issue 23, 1 December 2021, Pages 4593–4596, https://doi.org/10.1093/bioinformatics/btab665</a></p>
</blockquote>
<ul>
<li>We will use <code>MungeSumstats</code> here to reformat the summary statistics for Fluid Intelligence. The minimum info needed <em>after</em> munging is:<br>
</li>
<li>SNP, CHR, BP as first three columns.</li>
<li>It has at least one of these columns: (“Z”,“OR”,“BETA”,“LOG_ODDS”,“SIGNED_SUMSTAT”)</li>
</ul>
<p>Genome builds:</p>
<ul>
<li>The UK Biobank data from Ben Neale uses “GRCh37” so we will specify this.<br>
</li>
<li>However, if you don’t know which genome build you GWAS sum stats are in, <code>MungeSumstats</code> will infer this information by default (simply set <code>ref_genome=NULL</code>).</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path_formatted</span> <span class="op">&lt;-</span> <span class="fu">MungeSumstats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MungeSumstats/man/format_sumstats.html" class="external-link">format_sumstats</a></span><span class="op">(</span>path<span class="op">=</span><span class="va">gwas_sumstats_path</span>,
                                                 save_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".formatted.tsv.gz"</span><span class="op">)</span>,
                                                 ref_genome <span class="op">=</span><span class="st">"GRCh37"</span><span class="op">)</span></code></pre></div>
<p>For the sake of this example, we also provide a pre-munged version of the above file. If you use this, you can skip the previous two steps.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path_formatted</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_example_gwas.html">get_example_gwas</a></span><span class="op">(</span>trait <span class="op">=</span> <span class="st">"fluid_intelligence"</span>,
                                                     munged <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="map-snps-to-genes">Map SNPs to Genes<a class="anchor" aria-label="anchor" href="#map-snps-to-genes"></a>
</h3>
<p>Next, we must convert the GWAS summary statistics file to a gene-level signature so that we can compare it to a gene-level transcriptomic cell-type reference.</p>
<p>Note you can input the genome build of your summary statistics for this step or it can be inferred if left <code>NULL</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genesOutPath</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/map_snps_to_genes.html">map_snps_to_genes</a></span><span class="op">(</span>path_formatted <span class="op">=</span> <span class="va">path_formatted</span>,
                                                    genome_build <span class="op">=</span> <span class="st">"GRCh37"</span>,
                                                    genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="celltypedataset">CellTypeDataset<a class="anchor" aria-label="anchor" href="#celltypedataset"></a>
</h3>
<div class="section level4">
<h4 id="import-ctd">Import CTD<a class="anchor" aria-label="anchor" href="#import-ctd"></a>
</h4>
<p>The second step in <code>MAGMA.Celltyping</code> is to prepare your Cell Type Dataset (CTD) reference, from which we will get gene signatures for each cell-type in the reference.</p>
<p>The <a href="https://bioconductor.org/packages/release/data/experiment/html/ewceData.html" class="external-link"><code>ewceData</code></a> package comes with a CTD which we use as an example.</p>
<p>If you want to import your own single cell RNA-seq dataset, then this needs converting into CTD format; please see the <a href="https://nathanskene.github.io/EWCE/articles/extended.html#create-a-celltypedataset" class="external-link"><code>EWCE</code> vingette</a> for explanation of how to do this.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Note that the cell type dataset loaded in the code above is the Karolinksa cortex/hippocampus data only. For the full Karolinska dataset with hypothalamus and midbrain instead use the following:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_allKI"</span><span class="op">)</span></code></pre></div>
<p>You can find a list of other datasets provided by this package using <code><a href="../reference/get_ctd.html">?MAGMA.Celltyping::get_ctd</a></code>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Tasic"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DivSeq"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_AIBS"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DRONC_human"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DRONC_mouse"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_BlueLake2018_FrontalCortexOnly"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_BlueLake2018_VisualCortexOnly"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Saunders"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Zeisel2018"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="prepare-quantiles">Prepare quantiles<a class="anchor" aria-label="anchor" href="#prepare-quantiles"></a>
</h4>
<p>Preparing quantiles in the CTD is now something that is automatically handled by by <code>MAGMA.Celltyping</code> so the user doesn’t need to do this themselves as a pre-step anymore.</p>
<p>However, we still export the function <code>prepare_quantile_groups</code> in case users want to have more control.</p>
<p><code>prepare_quantile_groups</code> calculates the quantile groups for each celltype within the single-cell dataset. If your single-cell dataset is not from mouse, then change the <code>ctd_species</code> argument.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/prepare_quantile_groups.html">prepare_quantile_groups</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                                                 ctd_species<span class="op">=</span><span class="st">"mouse"</span><span class="op">)</span>
<span class="co"># Examine how the quantile groups look</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ctd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">specificity_quantiles</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gfap"</span>,<span class="st">"Dlg4"</span>,<span class="st">"Aif1"</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">ctd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">specificity_quantiles</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="run-the-main-cell-type-association-analysis">Run the main cell type association analysis<a class="anchor" aria-label="anchor" href="#run-the-main-cell-type-association-analysis"></a>
</h2>
<p>The analyses can be run in either linear or top10% enrichment modes. Each have their own advantages/disadvantages.</p>
<ul>
<li>
<strong>Linear mode</strong>: Performs a linear regression analysis using continuous gene-association scores from the GWAS MAGMA results and continuous specificity scores from each cell-type in the CTD reference. This mode takes advantage of gene-level weights in order to get more accurate association estimates. However, this mode also tends to give more significant results in general, as it may be capturing signatures from multiple cell-types at once.<br>
</li>
<li>
<strong>Top 10% mode</strong>: Performs gene-set enrichment analysis (GSEA) using only the top 10% most cell-type-specific genes for each cell-type (i.e. genes in the top specificity decile). Unlike linear mode, top 10% mode discards any gene-level weight data during testing and instead treats genes as categorical data (in the gene list or not in the gene list). While you do lose the nuance of the gene-level weights, you potential capture less cell-type signatures in a given test (due to only using a handful of top genes), which can make interpretation easier.</li>
</ul>
<p>Let’s start with linear:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctAssocsLinear</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_celltype_associations.html">calculate_celltype_associations</a></span><span class="op">(</span>
  ctd<span class="op">=</span><span class="va">ctd</span>,
  gwas_sumstats_path<span class="op">=</span><span class="va">gwas_sumstats_path_formatted</span>,
  genome_ref_path<span class="op">=</span><span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="va">FigsLinear</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>
  ctAssocs<span class="op">=</span><span class="va">ctAssocsLinear</span>,
  ctd<span class="op">=</span><span class="va">ctd</span><span class="op">)</span></code></pre></div>
<p>Now let’s add the top 10% mode</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctAssocsTop</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_celltype_associations.html">calculate_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path_formatted</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  EnrichmentMode <span class="op">=</span> <span class="st">"Top 10%"</span><span class="op">)</span>

<span class="va">FigsTopDecile</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>
  ctAssocs <span class="op">=</span> <span class="va">ctAssocsTop</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span></code></pre></div>
<p>Then plot linear together with the top 10% mode.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctAssocMerged</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_magma_results.html">merge_magma_results</a></span><span class="op">(</span>
  ctAssoc1 <span class="op">=</span> <span class="va">ctAssocsLinear</span>,
  ctAssoc2 <span class="op">=</span> <span class="va">ctAssocsTop</span><span class="op">)</span>

<span class="va">FigsMerged</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>
  ctAssocs <span class="op">=</span> <span class="va">ctAssocMerged</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span></code></pre></div>
<div class="section level4">
<h4 id="conditional-analysis-linear-mode">Conditional analysis (linear mode)<a class="anchor" aria-label="anchor" href="#conditional-analysis-linear-mode"></a>
</h4>
<p>Sometimes a single cell-type can dominate the signature in a given GWAS (e.g. pyramidal cells in Schizophrenia). This makes it difficult to disentangle whether other cell-types are truly associated with the GWAS trait, or whether the are simply enriched because they have an overlapping signature with the top cell-type. Conditional mode allows you to control for one (or several) cell-type(s) and see whether other cell-types still remain significantly enriched.</p>
<p>By default, it is assumed that you want to run the linear enrichment analysis. There are two modes for conditional analyses, you can either control for the top N cell types from the baseline analysis (in which case, set <code>controlTopNcells</code>) or control for specific specified cell types (in which case, set <code>controlledCTs</code>).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctCondAssocs</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_celltype_associations.html">calculate_conditional_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path_formatted <span class="op">=</span> <span class="va">path_formatted</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  analysis_name <span class="op">=</span> <span class="st">"Conditional"</span>,
  controlTopNcells<span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>ctAssocs <span class="op">=</span> <span class="va">ctCondAssocs</span>,
                                             ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span></code></pre></div>
<p>Let’s try as an alternative to control for expression of 3 neuronal subtypes at the same time:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctCondAssocs</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_celltype_associations.html">calculate_conditional_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">path_formatted</span>,
  genome_ref_path<span class="op">=</span><span class="va">genome_ref_path</span>,
  analysis_name <span class="op">=</span> <span class="st">"Conditional"</span>,
  controlledCTs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal CA1"</span>,<span class="st">"pyramidal SS"</span>,<span class="st">"interneurons"</span><span class="op">)</span>,
  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>ctAssocs <span class="op">=</span> <span class="va">ctCondAssocs</span>,
                                             ctd<span class="op">=</span><span class="va">ctd</span><span class="op">)</span></code></pre></div>
<p>Note that Periventricular Microglia (PVM) go from totally non-significant to significant once the neurons are controlled for. Test if this change is significant as follows:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">magma1</span> <span class="op">=</span> <span class="va">ctCondAssocs</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span><span class="op">[</span><span class="va">ctCondAssocs</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span><span class="op">$</span><span class="va">CONTROL</span><span class="op">==</span><span class="st">"BASELINE"</span>,<span class="op">]</span>
<span class="va">magma2</span> <span class="op">=</span> <span class="va">ctCondAssocs</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span><span class="op">[</span><span class="va">ctCondAssocs</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span><span class="op">$</span><span class="va">CONTROL</span><span class="op">==</span><span class="st">"pyramidal CA1,pyramidal SS,interneurons"</span>,<span class="op">]</span>

<span class="va">resCompared</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">:::</span><span class="fu"><a href="../reference/compare_trait_enrichments.html">compare_trait_enrichments</a></span><span class="op">(</span>magma1<span class="op">=</span><span class="va">magma1</span>,
                                                            magma2<span class="op">=</span><span class="va">magma2</span>,
                                                            annotLevel<span class="op">=</span><span class="fl">2</span>,
                                                            ctd<span class="op">=</span><span class="va">ctd</span><span class="op">)</span>
<span class="va">resCompared</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span></code></pre></div>
<p>Using this approach we can see that the increased enrichment is microglia in the controlled analysis is almost significantly increased relative to the baseline analysis.</p>
</div>
<div class="section level3">
<h3 id="conditional-analyses-top-10-mode">Conditional analyses (top 10% mode)<a class="anchor" aria-label="anchor" href="#conditional-analyses-top-10-mode"></a>
</h3>
<p>Conditional analyses can also be performed with top 10% mode (although the conditioning is done in linear mode).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctCondAssocsTopTen</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_celltype_associations.html">calculate_conditional_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">path_formatted</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  analysis_name <span class="op">=</span> <span class="st">"Conditional"</span>,
  controlledCTs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal CA1"</span>,<span class="st">"pyramidal SS"</span>,<span class="st">"interneurons"</span><span class="op">)</span>,
  controlledAnnotLevel<span class="op">=</span><span class="fl">1</span>,
  EnrichmentMode <span class="op">=</span> <span class="st">"Top 10%"</span><span class="op">)</span>

<span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>ctAssocs <span class="op">=</span> <span class="va">ctCondAssocsTopTen</span>,
                                             ctd<span class="op">=</span><span class="va">ctd</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="controlling-for-a-second-gwas">Controlling for a second GWAS<a class="anchor" aria-label="anchor" href="#controlling-for-a-second-gwas"></a>
</h3>
<p>We now want to test enrichments that remain in a GWAS after we control for a second GWAS. So let’s download a second GWAS sumstats file and prepare it for analysis.</p>
<p>20016.assoc.tsv is the sumstats file for ‘Fluid Intelligence Score’ from the UK Biobank.</p>
<p>So let’s subtract genes associated with prospective memory from those involved in fluid intelligence.</p>
<div class="section level4">
<h4 id="download-and-prepare-the-prospective-memory-gwas-summary-statistics">Download and prepare the ‘Prospective memory’ GWAS summary statistics<a class="anchor" aria-label="anchor" href="#download-and-prepare-the-prospective-memory-gwas-summary-statistics"></a>
</h4>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwas_sumstats_path</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_example_gwas.html">get_example_gwas</a></span><span class="op">(</span>storage_dir <span class="op">=</span> <span class="va">storage_dir</span>, 
                                                        trait <span class="op">=</span> <span class="st">"prospective_memory"</span><span class="op">)</span>
<span class="co"># Format sumstats - infer genome build from the data</span>
<span class="va">tmpSumStatsPath</span> <span class="op">=</span>  <span class="fu">MungeSumstats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MungeSumstats/man/format_sumstats.html" class="external-link">format_sumstats</a></span><span class="op">(</span>path<span class="op">=</span><span class="va">gwas_sumstats_path</span><span class="op">)</span>
<span class="va">path_formatted_memory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%s.formatted"</span>,<span class="va">gwas_sumstats_path</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span>from<span class="op">=</span><span class="va">tmpSumStatsPath</span>,
          to<span class="op">=</span><span class="va">path_formatted_memory</span>,
          overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#map SNPs to genes and infer the genome build fromt he data whilst doing so</span>
<span class="va">genesOutPath</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/map_snps_to_genes.html">map_snps_to_genes</a></span><span class="op">(</span>path_formatted <span class="op">=</span> <span class="va">path_formatted_memory</span>,
                                                   genome_ref_path<span class="op">=</span><span class="va">genome_ref_path</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="check-which-cell-types-this-gwas-is-associated-with-at-baseline">Check which cell types this GWAS is associated with at baseline<a class="anchor" aria-label="anchor" href="#check-which-cell-types-this-gwas-is-associated-with-at-baseline"></a>
</h4>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path_formatted_intelligence</span> <span class="op">&lt;-</span> <span class="va">path_formatted</span>

<span class="va">ctAssocsLinear_memory</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_celltype_associations.html">calculate_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">path_formatted_memory</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="va">ctAssocsLinear_intelligence</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_celltype_associations.html">calculate_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">path_formatted_intelligence</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>ctAssocs <span class="op">=</span> <span class="va">ctAssocsLinear_memory</span>,
                                             ctd<span class="op">=</span><span class="va">ctd</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="compare-enrichments-in-the-two-gwas-using-a-tile-plot">Compare enrichments in the two GWAS using a tile plot<a class="anchor" aria-label="anchor" href="#compare-enrichments-in-the-two-gwas-using-a-tile-plot"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctAssocMerged_MemInt</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_magma_results.html">merge_magma_results</a></span><span class="op">(</span>
  ctAssoc1 <span class="op">=</span> <span class="va">ctAssocsLinear_memory</span>,
  ctAssoc2 <span class="op">=</span> <span class="va">ctAssocsLinear_intelligence</span><span class="op">)</span>

<span class="va">FigsMerged_MemInt</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/magma_tileplot.html">magma_tileplot</a></span><span class="op">(</span>ctd<span class="op">=</span><span class="va">ctd</span>,
                                                      results<span class="op">=</span><span class="va">ctAssocMerged_MemInt</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span>,
                                                      annotLevel<span class="op">=</span><span class="fl">1</span>,
                                                      fileTag<span class="op">=</span><span class="st">"Merged_MemInt_lvl1"</span>,
                                                      output_path <span class="op">=</span> <span class="va">storage_dir</span><span class="op">)</span>

<span class="va">FigsMerged_MemInt</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/magma_tileplot.html">magma_tileplot</a></span><span class="op">(</span>ctd<span class="op">=</span><span class="va">ctd</span>,
                                                     results<span class="op">=</span><span class="va">ctAssocMerged_MemInt</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span>,
                                                     annotLevel<span class="op">=</span><span class="fl">2</span>,
                                                     fileTag<span class="op">=</span><span class="st">"Merged_MemInt_lvl2"</span>,
                                                     output_path <span class="op">=</span> <span class="va">storage_dir</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="check-which-cell-types-fluid-intelligence-is-associated-with-after-controlling-for-prospective-memory">Check which cell types ‘Fluid Intelligence’ is associated with after controlling for ‘Prospective memory’<a class="anchor" aria-label="anchor" href="#check-which-cell-types-fluid-intelligence-is-associated-with-after-controlling-for-prospective-memory"></a>
</h4>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Set paths for GWAS sum stats + .genes.out file (with the z-scores)</span>
<span class="va">gwas_sumstats_path</span> <span class="op">&lt;-</span> <span class="va">path_formatted_intelligence</span> <span class="co"># "/Users/natske/GWAS_Summary_Statistics/20016.assoc.tsv"</span>
<span class="va">memoryGenesOut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%s.genes.out"</span>,
                         <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_magma_paths.html">get_magma_paths</a></span><span class="op">(</span>
                           gwas_sumstats_path <span class="op">=</span> <span class="va">path_formatted_memory</span>,
                           upstream_kb <span class="op">=</span> <span class="fl">10</span>,
                           downstream_kb <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">$</span><span class="va">filePathPrefix</span><span class="op">)</span>

<span class="va">ctAssocsLinear</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_celltype_associations.html">calculate_celltype_associations</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span>,
  genesOutCOND <span class="op">=</span> <span class="va">memoryGenesOut</span>,
  analysis_name <span class="op">=</span> <span class="st">"ControllingForPropMemory"</span><span class="op">)</span>

<span class="va">FigsLinear</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype_associations.html">plot_celltype_associations</a></span><span class="op">(</span>ctAssocs <span class="op">=</span> <span class="va">ctAssocsLinear</span>,
                                                          ctd <span class="op">=</span> <span class="va">ctd</span>,
                                                          fileTag <span class="op">=</span> <span class="st">"ControllingForPropMemory"</span><span class="op">)</span></code></pre></div>
<p>We find that after controlling for prospective memory, there is no significant enrichment left associated with fluid intelligence.</p>
</div>
</div>
<div class="section level3">
<h3 id="calculate-cell-type-enrichments-directly-using-linear-model">Calculate cell type enrichments directly (using linear model)<a class="anchor" aria-label="anchor" href="#calculate-cell-type-enrichments-directly-using-linear-model"></a>
</h3>
<p>Instead of relying on the tool MAGMA (which requires an extra installation step), we can also compute cell-type enrichment scores in an R-native re-implementation of the MAGMA enrichment analyses using the following functions:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">magmaGenesOut</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/adjust_zstat_in_genesOut.html">adjust_zstat_in_genesOut</a></span><span class="op">(</span>
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  magma_file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"/Users/natske/GWAS_Summary_Statistics/MAGMA_Files"</span>,
                       <span class="st">"20016.assoc.tsv.10UP.1.5DOWN"</span>,
                       <span class="st">"20016.assoc.tsv.10UP.1.5DOWN.genes.out"</span><span class="op">)</span>,
  ctd_species<span class="op">=</span><span class="st">"mouse"</span><span class="op">)</span>

<span class="va">output</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_celltype_enrichment_limma.html">calculate_celltype_enrichment_limma</a></span><span class="op">(</span>
  magmaAdjZ <span class="op">=</span> <span class="va">magmaGenesOut</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  thresh <span class="op">=</span> <span class="fl">0.0001</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span>,
  annotLevel <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p>We can then get the probability of the celltype being enriched as follows</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The results should closely resemble those obtained using MAGMA</p>
</div>
<div class="section level3">
<h3 id="gene-set-enrichments">Gene set enrichments<a class="anchor" aria-label="anchor" href="#gene-set-enrichments"></a>
</h3>
<p>To test whether a gene set is enriched using MAGMA the following commands can be used:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"rbfox_binding"</span><span class="op">)</span>

<span class="va">gwas_sumstats_path</span> <span class="op">=</span> <span class="st">"/Users/natske/GWAS_Summary_Statistics/20016.assoc.tsv"</span>
<span class="va">geneset_res</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_geneset_enrichment.html">calculate_geneset_enrichment</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">rbfox_binding</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  analysis_name <span class="op">=</span> <span class="st">"Rbfox_20016"</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  geneset_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">geneset_res</span><span class="op">)</span></code></pre></div>
<p>We can then test whether the gene set is still enriched after controlling for celltype enrichment:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_allKI"</span><span class="op">)</span>

<span class="va">analysis_name</span> <span class="op">&lt;-</span> <span class="st">"Rbfox_16_pyrSS"</span>
<span class="va">controlledCTs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal SS"</span><span class="op">)</span>

<span class="va">cond_geneset_res_pyrSS</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_geneset_enrichment.html">calculate_conditional_geneset_enrichment</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">geneset</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,
  controlledCTs <span class="op">=</span> <span class="va">controlledCTs</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  analysis_name <span class="op">=</span> <span class="va">analysis_name</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="va">controlledCTs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal CA1"</span><span class="op">)</span>
<span class="va">cond_geneset_res_pyrCA1</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_geneset_enrichment.html">calculate_conditional_geneset_enrichment</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">geneset</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,
  controlledCTs <span class="op">=</span> <span class="va">controlledCTs</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  analysis_name <span class="op">=</span> <span class="va">analysis_name</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="va">controlledCTs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal CA1"</span>,<span class="st">"pyramidal SS"</span><span class="op">)</span>
<span class="va">cond_geneset_res_pyr</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_geneset_enrichment.html">calculate_conditional_geneset_enrichment</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">geneset</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,
  controlledCTs <span class="op">=</span> <span class="va">controlledCTs</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  analysis_name <span class="op">=</span> <span class="va">analysis_name</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="va">controlledCTs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Medium Spiny Neuron"</span><span class="op">)</span>
<span class="va">cond_geneset_res_MSN</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_geneset_enrichment.html">calculate_conditional_geneset_enrichment</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">geneset</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,
  controlledCTs <span class="op">=</span> <span class="va">controlledCTs</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  analysis_name <span class="op">=</span> <span class="va">analysis_name</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="va">controlledCTs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Medium Spiny Neuron"</span>,<span class="st">"pyramidal CA1"</span>,<span class="st">"pyramidal SS"</span>,<span class="st">"interneurons"</span><span class="op">)</span>
<span class="va">cond_geneset_res</span> <span class="op">=</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_conditional_geneset_enrichment.html">calculate_conditional_geneset_enrichment</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">geneset</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  controlledAnnotLevel <span class="op">=</span> <span class="fl">1</span>,
  controlledCTs <span class="op">=</span> <span class="va">controlledCTs</span>,
  gwas_sumstats_path <span class="op">=</span> <span class="va">gwas_sumstats_path</span>,
  analysis_name <span class="op">=</span> <span class="va">analysis_name</span>,
  genome_ref_path <span class="op">=</span> <span class="va">genome_ref_path</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R Under development (unstable) (2022-01-29 r81593)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] dplyr_1.0.7            ggplot2_3.3.5          MAGMA.Celltyping_2.0.0</span>
<span class="co">## [4] BiocStyle_2.23.1      </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.1               AnnotationHub_3.3.8          </span>
<span class="co">##   [3] BiocFileCache_2.3.4           systemfonts_1.0.3            </span>
<span class="co">##   [5] googleAuthR_2.0.0             plyr_1.8.6                   </span>
<span class="co">##   [7] lazyeval_0.2.2                splines_4.2.0                </span>
<span class="co">##   [9] orthogene_1.1.1               ewceData_1.3.0               </span>
<span class="co">##  [11] BiocParallel_1.29.12          GenomeInfoDb_1.31.4          </span>
<span class="co">##  [13] digest_0.6.29                 htmltools_0.5.2              </span>
<span class="co">##  [15] RNOmni_1.0.0                  fansi_1.0.2                  </span>
<span class="co">##  [17] magrittr_2.0.2                memoise_2.0.1                </span>
<span class="co">##  [19] BSgenome_1.63.4               limma_3.51.3                 </span>
<span class="co">##  [21] Biostrings_2.63.1             matrixStats_0.61.0           </span>
<span class="co">##  [23] R.utils_2.11.0                pkgdown_2.0.2.9000           </span>
<span class="co">##  [25] prettyunits_1.1.1             colorspace_2.0-2             </span>
<span class="co">##  [27] blob_1.2.2                    rappdirs_0.3.3               </span>
<span class="co">##  [29] textshaping_0.3.6             xfun_0.29                    </span>
<span class="co">##  [31] crayon_1.4.2                  RCurl_1.98-1.5               </span>
<span class="co">##  [33] jsonlite_1.7.3                lme4_1.1-27.1                </span>
<span class="co">##  [35] VariantAnnotation_1.41.3      glue_1.6.1                   </span>
<span class="co">##  [37] gargle_1.2.0                  gtable_0.3.0                 </span>
<span class="co">##  [39] zlibbioc_1.41.0               XVector_0.35.0               </span>
<span class="co">##  [41] HGNChelper_0.8.1              DelayedArray_0.21.2          </span>
<span class="co">##  [43] car_3.0-12                    SingleCellExperiment_1.17.2  </span>
<span class="co">##  [45] BiocGenerics_0.41.2           abind_1.4-5                  </span>
<span class="co">##  [47] scales_1.1.1                  DBI_1.1.2                    </span>
<span class="co">##  [49] rstatix_0.7.0                 Rcpp_1.0.8                   </span>
<span class="co">##  [51] progress_1.2.2                viridisLite_0.4.0            </span>
<span class="co">##  [53] xtable_1.8-4                  bit_4.0.4                    </span>
<span class="co">##  [55] stats4_4.2.0                  htmlwidgets_1.5.4            </span>
<span class="co">##  [57] httr_1.4.2                    ellipsis_0.3.2               </span>
<span class="co">##  [59] pkgconfig_2.0.3               XML_3.99-0.8                 </span>
<span class="co">##  [61] R.methodsS3_1.8.1             sass_0.4.0                   </span>
<span class="co">##  [63] dbplyr_2.1.1                  utf8_1.2.2                   </span>
<span class="co">##  [65] tidyselect_1.1.1              rlang_1.0.0                  </span>
<span class="co">##  [67] reshape2_1.4.4                later_1.3.0                  </span>
<span class="co">##  [69] AnnotationDbi_1.57.1          munsell_0.5.0                </span>
<span class="co">##  [71] BiocVersion_3.15.0            tools_4.2.0                  </span>
<span class="co">##  [73] cachem_1.0.6                  cli_3.1.1                    </span>
<span class="co">##  [75] generics_0.1.2                RSQLite_2.2.9                </span>
<span class="co">##  [77] ExperimentHub_2.3.5           MungeSumstats_1.3.1          </span>
<span class="co">##  [79] broom_0.7.12                  ggdendro_0.1.22              </span>
<span class="co">##  [81] evaluate_0.14                 stringr_1.4.0                </span>
<span class="co">##  [83] fastmap_1.1.0                 yaml_2.2.2                   </span>
<span class="co">##  [85] ragg_1.2.1                    babelgene_21.4               </span>
<span class="co">##  [87] knitr_1.37                    bit64_4.0.5                  </span>
<span class="co">##  [89] fs_1.5.2                      purrr_0.3.4                  </span>
<span class="co">##  [91] KEGGREST_1.35.0               gprofiler2_0.2.1             </span>
<span class="co">##  [93] nlme_3.1-155                  mime_0.12                    </span>
<span class="co">##  [95] R.oo_1.24.0                   xml2_1.3.3                   </span>
<span class="co">##  [97] biomaRt_2.51.2                compiler_4.2.0               </span>
<span class="co">##  [99] plotly_4.10.0                 filelock_1.0.2               </span>
<span class="co">## [101] curl_4.3.2                    png_0.1-7                    </span>
<span class="co">## [103] interactiveDisplayBase_1.33.0 ggsignif_0.6.3               </span>
<span class="co">## [105] tibble_3.1.6                  EWCE_1.3.3                   </span>
<span class="co">## [107] bslib_0.3.1                   homologene_1.4.68.19.3.27    </span>
<span class="co">## [109] stringi_1.7.6                 GenomicFeatures_1.47.7       </span>
<span class="co">## [111] desc_1.4.0                    lattice_0.20-45              </span>
<span class="co">## [113] Matrix_1.4-0                  nloptr_2.0.0                 </span>
<span class="co">## [115] vctrs_0.3.8                   pillar_1.7.0                 </span>
<span class="co">## [117] lifecycle_1.0.1               BiocManager_1.30.16          </span>
<span class="co">## [119] jquerylib_0.1.4               data.table_1.14.2            </span>
<span class="co">## [121] bitops_1.0-7                  httpuv_1.6.5                 </span>
<span class="co">## [123] patchwork_1.1.1               rtracklayer_1.55.3           </span>
<span class="co">## [125] GenomicRanges_1.47.6          R6_2.5.1                     </span>
<span class="co">## [127] BiocIO_1.5.0                  bookdown_0.24                </span>
<span class="co">## [129] promises_1.2.0.1              gridExtra_2.3                </span>
<span class="co">## [131] IRanges_2.29.1                boot_1.3-28                  </span>
<span class="co">## [133] MASS_7.3-55                   assertthat_0.2.1             </span>
<span class="co">## [135] SummarizedExperiment_1.25.3   rprojroot_2.0.2              </span>
<span class="co">## [137] rjson_0.2.21                  withr_2.4.3                  </span>
<span class="co">## [139] GenomicAlignments_1.31.2      Rsamtools_2.11.0             </span>
<span class="co">## [141] S4Vectors_0.33.10             GenomeInfoDbData_1.2.7       </span>
<span class="co">## [143] hms_1.1.1                     parallel_4.2.0               </span>
<span class="co">## [145] grid_4.2.0                    minqa_1.2.4                  </span>
<span class="co">## [147] tidyr_1.2.0                   rmarkdown_2.11               </span>
<span class="co">## [149] MatrixGenerics_1.7.0          carData_3.0-5                </span>
<span class="co">## [151] ggpubr_0.4.0                  Biobase_2.55.0               </span>
<span class="co">## [153] shiny_1.7.1                   restfulr_0.0.13</span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
