<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MAGMA_Celltyping: Getting started • MAGMA.Celltyping</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MAGMA_Celltyping: Getting started">
<meta property="og:description" content="MAGMA.Celltyping">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MAGMA.Celltyping.html">MAGMA_Celltyping: Getting started</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">MAGMA_Celltyping: full_workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MAGMA_Celltyping: Getting started</h1>
                        <h4 data-toc-skip class="author"><h4>
Authors: <i>Brian M. Schilder, Alan Murphy, Julien Bryois &amp; Nathan Skene</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: <i>Feb-04-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/vignettes/MAGMA.Celltyping.Rmd" class="external-link"><code>vignettes/MAGMA.Celltyping.Rmd</code></a></small>
      <div class="hidden name"><code>MAGMA.Celltyping.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/MAGMA_Celltyping" class="external-link">MAGMA.Celltyping</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p><code>MAGMA.Celltyping</code> is a software package that facilitates conducting cell-type-specific enrichment tests on GWAS summary statistics.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Specify where you want the large files to be downloaded to.</p>
<p><em>NOTE</em>: Make sure you change <code>storage_dir</code> to somewhere other than <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code> if you want to make sure the results aren’t deleted after this R session closes!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">storage_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-data">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h2>
<div class="section level3">
<h3 id="gwas">GWAS<a class="anchor" aria-label="anchor" href="#gwas"></a>
</h3>
<ul>
<li>We need to have a summary statistics file to analyse as input.<br>
</li>
<li>As an example, you can download UK Biobank summary statistics for ‘fluid_intelligence’ using <code><a href="../reference/get_example_gwas.html">get_example_gwas()</a></code>.</li>
</ul>
<p>Here we provide a pre-munged version of the above file.</p>
<div class="section level4">
<h4 id="munging">Munging<a class="anchor" aria-label="anchor" href="#munging"></a>
</h4>
<p>Our lab have created <a href="https://github.com/neurogenomics/MungeSumstats" class="external-link"><code>MungeSumstats</code></a>, a robust Bioconductor package for formatting multiple types of summary statistics files. We highly recommend processing your GWAS summary statistics with <code>MungeSumstats</code> before continuing. See the <em>full_workflow</em> vignette for more details.</p>
<p>The minimum info needed <em>after</em> munging is:<br>
- “SNP”, “CHR”, and “BP” as first three columns. - It has at least one of these columns: “Z”,“OR”,“BETA”,“LOG_ODDS”,“SIGNED_SUMSTAT”</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path_formatted</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_example_gwas.html">get_example_gwas</a></span><span class="op">(</span>
  trait <span class="op">=</span> <span class="st">"prospective_memory"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="map-snps-to-genes">Map SNPs to Genes<a class="anchor" aria-label="anchor" href="#map-snps-to-genes"></a>
</h4>
<p>Note you can input the genome build of your summary statistics for this step or it can be inferred if left <code>NULL</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genesOutPath</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/map_snps_to_genes.html">map_snps_to_genes</a></span><span class="op">(</span>
  path_formatted <span class="op">=</span> <span class="va">path_formatted</span>,
  genome_build <span class="op">=</span> <span class="st">"GRCh37"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="magma_files_public">MAGMA_Files_Public<a class="anchor" aria-label="anchor" href="#magma_files_public"></a>
</h4>
<p>Rather than preprocessing the GWAS yourself, you can instead use the <a href="https://github.com/neurogenomics/MAGMA_Files_Public" class="external-link"><code>MAGMA_Files_Public</code></a> database we have created. It contains pre-computed MAGMA SNP-to-genes mapping files for hundreds of GWAS.</p>
<p>You can browse which GWAS traits are available by looking at the provided <a href="https://github.com/neurogenomics/MAGMA_Files_Public/blob/master/metadata.csv" class="external-link"><em>metadata.csv</em></a> file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">magma_dirs</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/import_magma_files.html">import_magma_files</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="st">"ieu-a-298"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using built-in example files: ieu-a-298.tsv.gz.35UP.10DOWN</span></code></pre>
<pre><code><span class="co">## Returning MAGMA directories.</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="celltypedataset">CellTypeDataset<a class="anchor" aria-label="anchor" href="#celltypedataset"></a>
</h3>
<p><code>ewceData</code> provides a number of CellTypeDatasets (CTD) to be used a cell-type transcriptomic signature reference files.</p>
<p>If you want to create your own single-cell transcriptomic reference, you’ll need to first convert it to CTD using the instructions found in the <code>EWCE</code> package <a href="https://github.com/NathanSkene/EWCE/" class="external-link">documentation here</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## snapshotDate(): 2022-01-20</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<p>Note that the cell type dataset loaded in the code above is the Karolinksa cortex/hippocampus data only. For the full Karolinska dataset with hypothalamus and midbrain instead use the following:</p>
<pre><code><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_allKI"</span><span class="op">)</span></code></pre>
<p>Or for the DRONC seq or AIBS datasets use:</p>
<pre><code><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Tasic"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DivSeq"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_AIBS"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DRONC_human"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DRONC_mouse"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_BlueLake2018_FrontalCortexOnly"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_BlueLake2018_VisualCortexOnly"</span><span class="op">)</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Saunders"</span><span class="op">)</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="run-cell-type-enrichment-analyses">Run cell-type enrichment analyses<a class="anchor" aria-label="anchor" href="#run-cell-type-enrichment-analyses"></a>
</h2>
<p><code>MAGMA.Celltyping</code> offers a suite of functions for conducting various types of cell-type-specific enrichment tests on GWAS summary statistics.</p>
<p>The <code>celltype_associations_pipeline</code> wraps several functions that in previous versions of <code>MAGMA.Celltyping</code> had to be set up and run separately. These include:</p>
<ul>
<li>
<strong>Linear enrichment</strong>: <code>calculate_celltype_associations(EnrichmentMode = "linear")</code> internally. Activated when <code>run_linear=TRUE</code>.</li>
<li>
<strong>Top 10% enrichment</strong>: Uses <code>calculate_celltype_associations(EnrichmentMode = "Top 10%")</code> internally. Activated when <code>run_top10=TRUE</code>.</li>
<li>
<strong>Conditional enrichment</strong>: Uses <code>calculate_conditional_celltype_associations</code> internally. Activated when <code>run_conditional=TRUE</code>.</li>
</ul>
<p>Thus, <code>celltype_associations_pipeline</code> is designed to make these analyses easier to run.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MAGMA_results</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/celltype_associations_pipeline.html">celltype_associations_pipeline</a></span><span class="op">(</span>
  magma_dirs <span class="op">=</span> <span class="va">magma_dirs</span>,
  ctd <span class="op">=</span> <span class="va">ctd</span>,
  ctd_species <span class="op">=</span> <span class="st">"mouse"</span>, 
  ctd_name <span class="op">=</span> <span class="st">"Zeisel2015"</span>, 
  run_linear <span class="op">=</span> <span class="cn">TRUE</span>, 
  run_top10 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>We’ve also saved a pre-computed version of these results as a dataset:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MAGMA_results</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="va"><a href="../reference/enrichment_results.html">enrichment_results</a></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plot-results">Plot results<a class="anchor" aria-label="anchor" href="#plot-results"></a>
</h2>
<div class="section level3">
<h3 id="merge-results">Merge results<a class="anchor" aria-label="anchor" href="#merge-results"></a>
</h3>
<p><code>merge_results</code> imports each of the MAGMA enrichment results files and merges them into one so that they can easily be plotted and further analysed.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">merged_results</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_results.html">merge_results</a></span><span class="op">(</span>
  MAGMA_results <span class="op">=</span> <span class="va">MAGMA_results</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Saving full merged results to ==&gt; /tmp/Rtmp3qumIa/MAGMA_celltyping..lvl1.csv</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">merged_results</span><span class="op">)</span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="10%">
<col width="13%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="2%">
<col width="2%">
<col width="21%">
<col width="3%">
<col width="4%">
<col width="3%">
<col width="4%">
<col width="5%">
<col width="3%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">GWAS</th>
<th align="left">Celltype</th>
<th align="left">OBS_GENES</th>
<th align="right">BETA</th>
<th align="right">BETA_STD</th>
<th align="right">SE</th>
<th align="right">P</th>
<th align="right">level</th>
<th align="left">Method</th>
<th align="left">GCOV_FILE</th>
<th align="left">CONTROL</th>
<th align="left">CONTROL_label</th>
<th align="right">log10p</th>
<th align="left">genesOutCOND</th>
<th align="left">EnrichmentMode</th>
<th align="right">FDR</th>
<th align="left">Celltype_id</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">11809</td>
<td align="right">0.0012211</td>
<td align="right">0.0143570</td>
<td align="right">0.0006699</td>
<td align="right">0.034171</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.4663423</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.1943643</td>
<td align="left">interneurons</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">11809</td>
<td align="right">0.0010366</td>
<td align="right">0.0120100</td>
<td align="right">0.0006865</td>
<td align="right">0.065539</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.1835002</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">11809</td>
<td align="right">0.0009264</td>
<td align="right">0.0108780</td>
<td align="right">0.0006697</td>
<td align="right">0.083299</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.0793602</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_SS</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">260</td>
<td align="right">0.1015100</td>
<td align="right">0.0119480</td>
<td align="right">0.0512870</td>
<td align="right">0.023904</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.6215294</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.1943643</td>
<td align="left">microglia</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">290</td>
<td align="right">0.0713330</td>
<td align="right">0.0088599</td>
<td align="right">0.0484960</td>
<td align="right">0.070669</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.1507711</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">289</td>
<td align="right">0.0764730</td>
<td align="right">0.0094821</td>
<td align="right">0.0487460</td>
<td align="right">0.058356</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.2339145</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_SS</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">11809</td>
<td align="right">0.0012211</td>
<td align="right">0.0143570</td>
<td align="right">0.0006699</td>
<td align="right">0.034171</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.4663423</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.1943643</td>
<td align="left">interneurons</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">11809</td>
<td align="right">0.0010366</td>
<td align="right">0.0120100</td>
<td align="right">0.0006865</td>
<td align="right">0.065539</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.1835002</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">11809</td>
<td align="right">0.0009264</td>
<td align="right">0.0108780</td>
<td align="right">0.0006697</td>
<td align="right">0.083299</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.0793602</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_SS</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">260</td>
<td align="right">0.1015100</td>
<td align="right">0.0119480</td>
<td align="right">0.0512870</td>
<td align="right">0.023904</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.6215294</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.1943643</td>
<td align="left">microglia</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">290</td>
<td align="right">0.0713330</td>
<td align="right">0.0088599</td>
<td align="right">0.0484960</td>
<td align="right">0.070669</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.1507711</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">289</td>
<td align="right">0.0764730</td>
<td align="right">0.0094821</td>
<td align="right">0.0487460</td>
<td align="right">0.058356</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-1.2339145</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.1943643</td>
<td align="left">pyramidal_SS</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">293</td>
<td align="right">0.0311950</td>
<td align="right">0.0038943</td>
<td align="right">0.0497550</td>
<td align="right">0.265340</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.5761973</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.5306800</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">293</td>
<td align="right">0.0311950</td>
<td align="right">0.0038943</td>
<td align="right">0.0497550</td>
<td align="right">0.265340</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.5761973</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.5306800</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">284</td>
<td align="right">0.0159080</td>
<td align="right">0.0019556</td>
<td align="right">0.0492120</td>
<td align="right">0.373250</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.4280002</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.6531875</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">284</td>
<td align="right">0.0159080</td>
<td align="right">0.0019556</td>
<td align="right">0.0492120</td>
<td align="right">0.373250</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.4280002</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.6531875</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">11809</td>
<td align="right">-0.0000893</td>
<td align="right">-0.0010659</td>
<td align="right">0.0006559</td>
<td align="right">0.554160</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.2563648</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.7638867</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">11809</td>
<td align="right">-0.0002481</td>
<td align="right">-0.0030903</td>
<td align="right">0.0006231</td>
<td align="right">0.654760</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.1839179</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.7638867</td>
<td align="left">microglia</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">11809</td>
<td align="right">-0.0001820</td>
<td align="right">-0.0021308</td>
<td align="right">0.0006861</td>
<td align="right">0.604610</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.2185247</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.7638867</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">282</td>
<td align="right">-0.0127510</td>
<td align="right">-0.0015621</td>
<td align="right">0.0481600</td>
<td align="right">0.604410</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.2186684</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.7638867</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">11809</td>
<td align="right">-0.0000893</td>
<td align="right">-0.0010659</td>
<td align="right">0.0006559</td>
<td align="right">0.554160</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.2563648</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.7638867</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">11809</td>
<td align="right">-0.0002481</td>
<td align="right">-0.0030903</td>
<td align="right">0.0006231</td>
<td align="right">0.654760</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.1839179</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.7638867</td>
<td align="left">microglia</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">11809</td>
<td align="right">-0.0001820</td>
<td align="right">-0.0021308</td>
<td align="right">0.0006861</td>
<td align="right">0.604610</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.2185247</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.7638867</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">282</td>
<td align="right">-0.0127510</td>
<td align="right">-0.0015621</td>
<td align="right">0.0481600</td>
<td align="right">0.604410</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.2186684</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.7638867</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">11809</td>
<td align="right">-0.0006254</td>
<td align="right">-0.0075750</td>
<td align="right">0.0006451</td>
<td align="right">0.833860</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.0789069</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.8980031</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">11809</td>
<td align="right">-0.0006254</td>
<td align="right">-0.0075750</td>
<td align="right">0.0006451</td>
<td align="right">0.833860</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.0789069</td>
<td align="left">NA</td>
<td align="left">Linear</td>
<td align="right">0.8980031</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">287</td>
<td align="right">-0.0645100</td>
<td align="right">-0.0079715</td>
<td align="right">0.0472000</td>
<td align="right">0.914140</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.0389873</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.9141400</td>
<td align="left">interneurons</td>
</tr>
<tr class="even">
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">287</td>
<td align="right">-0.0645100</td>
<td align="right">-0.0079715</td>
<td align="right">0.0472000</td>
<td align="right">0.914140</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">finn-a-AD.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="right">-0.0389873</td>
<td align="left">NA</td>
<td align="left">Top 10%</td>
<td align="right">0.9141400</td>
<td align="left">interneurons</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>Now we’ll construct a heatmap visualizing the enrichment results, such that each GWAS is shown on the y-axis and each cell-type is shown on the x-axis. Results can be further facetted by what kind of test was run (linear, top10%, and/or conditional).</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">heat</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/results_heatmap.html">results_heatmap</a></span><span class="op">(</span>
  merged_results <span class="op">=</span> <span class="va">merged_results</span>, 
  title <span class="op">=</span> <span class="st">"Alzheimer's Disease (ieu-a-298) vs. nervous system cell-types (Zeisel2015)"</span>,
  fdr_thresh <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 28 results @ FDR &lt; 1</span></code></pre>
<p><img src="MAGMA.Celltyping_files/figure-html/Heatmap-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="top-results">Top results<a class="anchor" aria-label="anchor" href="#top-results"></a>
</h2>
<div class="section level3">
<h3 id="top-phenotypes">Top phenotypes<a class="anchor" aria-label="anchor" href="#top-phenotypes"></a>
</h3>
<p>Get the phenotypes with the greatest number of significant cell-type enrichment results.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top_phenos</span> <span class="op">&lt;-</span> <span class="va">merged_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">EnrichmentMode</span>, <span class="va">GWAS</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Celltype<span class="op">=</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">Celltype</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">Celltype</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `summarise()` has grouped output by 'EnrichmentMode'. You can override using the</span>
<span class="co">## `.groups` argument.</span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">top_phenos</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 2 × 3</span></span>
<span class="co">## <span style="color: #949494;"># Groups:   EnrichmentMode [2]</span></span>
<span class="co">##   EnrichmentMode GWAS                         Celltype</span>
<span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;">1</span> Linear         finn-a-AD.tsv.gz.35UP.10DOWN        7</span>
<span class="co">## <span style="color: #BCBCBC;">2</span> Top 10%        finn-a-AD.tsv.gz.35UP.10DOWN        7</span></code></pre>
</div>
<div class="section level3">
<h3 id="top-enrichments">Top enrichments<a class="anchor" aria-label="anchor" href="#top-enrichments"></a>
</h3>
<p>Get the phenotypes-celltype enrichment results with the most significant p-values (per phenotype).</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top_enrich</span> <span class="op">&lt;-</span> <span class="va">merged_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">EnrichmentMode</span>, <span class="va">GWAS</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_min</a></span><span class="op">(</span><span class="va">FDR</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">top_enrich</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 12 × 17</span></span>
<span class="co">## <span style="color: #949494;"># Groups:   EnrichmentMode, GWAS [2]</span></span>
<span class="co">##    GWAS          Celltype OBS_GENES    BETA BETA_STD      SE      P level Method</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> finn-a-AD.ts… <span style="color: #949494;">"</span> i n t… 11809     1.22<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>  0.014<span style="text-decoration: underline;">4</span>  6.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.034<span style="text-decoration: underline;">2</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 11809     1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>  0.012<span style="text-decoration: underline;">0</span>  6.86<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.065<span style="text-decoration: underline;">5</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 11809     9.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>  0.010<span style="text-decoration: underline;">9</span>  6.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.083<span style="text-decoration: underline;">3</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> finn-a-AD.ts… <span style="color: #949494;">"</span> i n t… 11809     1.22<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>  0.014<span style="text-decoration: underline;">4</span>  6.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.034<span style="text-decoration: underline;">2</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 11809     1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>  0.012<span style="text-decoration: underline;">0</span>  6.86<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.065<span style="text-decoration: underline;">5</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 11809     9.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>  0.010<span style="text-decoration: underline;">9</span>  6.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.083<span style="text-decoration: underline;">3</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> finn-a-AD.ts… <span style="color: #949494;">"</span> m i c… 260       1.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span>  0.011<span style="text-decoration: underline;">9</span>  5.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.023<span style="text-decoration: underline;">9</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 290       7.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>  0.008<span style="text-decoration: underline;">86</span> 4.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.070<span style="text-decoration: underline;">7</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 289       7.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>  0.009<span style="text-decoration: underline;">48</span> 4.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.058<span style="text-decoration: underline;">4</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;">10</span> finn-a-AD.ts… <span style="color: #949494;">"</span> m i c… 260       1.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span>  0.011<span style="text-decoration: underline;">9</span>  5.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.023<span style="text-decoration: underline;">9</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;">11</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 290       7.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>  0.008<span style="text-decoration: underline;">86</span> 4.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.070<span style="text-decoration: underline;">7</span>     1 MAGMA </span>
<span class="co">## <span style="color: #BCBCBC;">12</span> finn-a-AD.ts… <span style="color: #949494;">"</span> p y r… 289       7.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>  0.009<span style="text-decoration: underline;">48</span> 4.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.058<span style="text-decoration: underline;">4</span>     1 MAGMA </span>
<span class="co">## <span style="color: #949494;"># … with 8 more variables: GCOV_FILE &lt;chr&gt;, CONTROL &lt;chr&gt;, CONTROL_label &lt;chr&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   log10p &lt;dbl&gt;, genesOutCOND &lt;chr&gt;, EnrichmentMode &lt;chr&gt;, FDR &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   Celltype_id &lt;chr&gt;</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R Under development (unstable) (2022-01-29 r81593)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] ewceData_1.3.0         ExperimentHub_2.3.5    AnnotationHub_3.3.8   </span>
<span class="co">##  [4] BiocFileCache_2.3.4    dbplyr_2.1.1           BiocGenerics_0.41.2   </span>
<span class="co">##  [7] dplyr_1.0.7            ggplot2_3.3.5          MAGMA.Celltyping_2.0.1</span>
<span class="co">## [10] BiocStyle_2.23.1      </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.1               systemfonts_1.0.3            </span>
<span class="co">##   [3] googleAuthR_2.0.0             plyr_1.8.6                   </span>
<span class="co">##   [5] lazyeval_0.2.2                splines_4.2.0                </span>
<span class="co">##   [7] orthogene_1.1.1               BiocParallel_1.29.12         </span>
<span class="co">##   [9] GenomeInfoDb_1.31.4           digest_0.6.29                </span>
<span class="co">##  [11] htmltools_0.5.2               RNOmni_1.0.0                 </span>
<span class="co">##  [13] fansi_1.0.2                   magrittr_2.0.2               </span>
<span class="co">##  [15] memoise_2.0.1                 BSgenome_1.63.4              </span>
<span class="co">##  [17] limma_3.51.3                  Biostrings_2.63.1            </span>
<span class="co">##  [19] matrixStats_0.61.0            R.utils_2.11.0               </span>
<span class="co">##  [21] pkgdown_2.0.2.9000            prettyunits_1.1.1            </span>
<span class="co">##  [23] colorspace_2.0-2              blob_1.2.2                   </span>
<span class="co">##  [25] rappdirs_0.3.3                textshaping_0.3.6            </span>
<span class="co">##  [27] xfun_0.29                     crayon_1.4.2                 </span>
<span class="co">##  [29] RCurl_1.98-1.5                jsonlite_1.7.3               </span>
<span class="co">##  [31] lme4_1.1-27.1                 VariantAnnotation_1.41.3     </span>
<span class="co">##  [33] glue_1.6.1                    gargle_1.2.0                 </span>
<span class="co">##  [35] gtable_0.3.0                  zlibbioc_1.41.0              </span>
<span class="co">##  [37] XVector_0.35.0                HGNChelper_0.8.1             </span>
<span class="co">##  [39] DelayedArray_0.21.2           car_3.0-12                   </span>
<span class="co">##  [41] SingleCellExperiment_1.17.2   abind_1.4-5                  </span>
<span class="co">##  [43] scales_1.1.1                  DBI_1.1.2                    </span>
<span class="co">##  [45] rstatix_0.7.0                 Rcpp_1.0.8                   </span>
<span class="co">##  [47] progress_1.2.2                viridisLite_0.4.0            </span>
<span class="co">##  [49] xtable_1.8-4                  bit_4.0.4                    </span>
<span class="co">##  [51] stats4_4.2.0                  htmlwidgets_1.5.4            </span>
<span class="co">##  [53] httr_1.4.2                    ellipsis_0.3.2               </span>
<span class="co">##  [55] farver_2.1.0                  pkgconfig_2.0.3              </span>
<span class="co">##  [57] XML_3.99-0.8                  R.methodsS3_1.8.1            </span>
<span class="co">##  [59] sass_0.4.0                    utf8_1.2.2                   </span>
<span class="co">##  [61] labeling_0.4.2                tidyselect_1.1.1             </span>
<span class="co">##  [63] rlang_1.0.1                   reshape2_1.4.4               </span>
<span class="co">##  [65] later_1.3.0                   AnnotationDbi_1.57.1         </span>
<span class="co">##  [67] munsell_0.5.0                 BiocVersion_3.15.0           </span>
<span class="co">##  [69] tools_4.2.0                   cachem_1.0.6                 </span>
<span class="co">##  [71] cli_3.1.1                     generics_0.1.2               </span>
<span class="co">##  [73] RSQLite_2.2.9                 MungeSumstats_1.3.1          </span>
<span class="co">##  [75] broom_0.7.12                  ggdendro_0.1.22              </span>
<span class="co">##  [77] evaluate_0.14                 stringr_1.4.0                </span>
<span class="co">##  [79] fastmap_1.1.0                 yaml_2.2.2                   </span>
<span class="co">##  [81] ragg_1.2.1                    babelgene_21.4               </span>
<span class="co">##  [83] knitr_1.37                    bit64_4.0.5                  </span>
<span class="co">##  [85] fs_1.5.2                      purrr_0.3.4                  </span>
<span class="co">##  [87] KEGGREST_1.35.0               gprofiler2_0.2.1             </span>
<span class="co">##  [89] nlme_3.1-155                  mime_0.12                    </span>
<span class="co">##  [91] R.oo_1.24.0                   xml2_1.3.3                   </span>
<span class="co">##  [93] biomaRt_2.51.3                compiler_4.2.0               </span>
<span class="co">##  [95] plotly_4.10.0                 filelock_1.0.2               </span>
<span class="co">##  [97] curl_4.3.2                    png_0.1-7                    </span>
<span class="co">##  [99] interactiveDisplayBase_1.33.0 ggsignif_0.6.3               </span>
<span class="co">## [101] tibble_3.1.6                  EWCE_1.3.3                   </span>
<span class="co">## [103] bslib_0.3.1                   homologene_1.4.68.19.3.27    </span>
<span class="co">## [105] stringi_1.7.6                 highr_0.9                    </span>
<span class="co">## [107] GenomicFeatures_1.47.7        desc_1.4.0                   </span>
<span class="co">## [109] lattice_0.20-45               Matrix_1.4-0                 </span>
<span class="co">## [111] nloptr_2.0.0                  vctrs_0.3.8                  </span>
<span class="co">## [113] pillar_1.7.0                  lifecycle_1.0.1              </span>
<span class="co">## [115] BiocManager_1.30.16           jquerylib_0.1.4              </span>
<span class="co">## [117] data.table_1.14.2             bitops_1.0-7                 </span>
<span class="co">## [119] httpuv_1.6.5                  patchwork_1.1.1              </span>
<span class="co">## [121] rtracklayer_1.55.3            GenomicRanges_1.47.6         </span>
<span class="co">## [123] R6_2.5.1                      BiocIO_1.5.0                 </span>
<span class="co">## [125] bookdown_0.24                 promises_1.2.0.1             </span>
<span class="co">## [127] gridExtra_2.3                 IRanges_2.29.1               </span>
<span class="co">## [129] boot_1.3-28                   MASS_7.3-55                  </span>
<span class="co">## [131] assertthat_0.2.1              SummarizedExperiment_1.25.3  </span>
<span class="co">## [133] rprojroot_2.0.2               rjson_0.2.21                 </span>
<span class="co">## [135] withr_2.4.3                   GenomicAlignments_1.31.2     </span>
<span class="co">## [137] Rsamtools_2.11.0              S4Vectors_0.33.10            </span>
<span class="co">## [139] GenomeInfoDbData_1.2.7        hms_1.1.1                    </span>
<span class="co">## [141] parallel_4.2.0                grid_4.2.0                   </span>
<span class="co">## [143] minqa_1.2.4                   tidyr_1.2.0                  </span>
<span class="co">## [145] rmarkdown_2.11                MatrixGenerics_1.7.0         </span>
<span class="co">## [147] carData_3.0-5                 ggpubr_0.4.0                 </span>
<span class="co">## [149] Biobase_2.55.0                shiny_1.7.1                  </span>
<span class="co">## [151] restfulr_0.0.13</span></code></pre>
</details><hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
