<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started • MAGMA.Celltyping</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting started">
<meta property="og:description" content="MAGMA.Celltyping">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAGMA.Celltyping</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MAGMA.Celltyping.html">Getting started</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/full_workflow.html">Full workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/MAGMA_Celltyping/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started</h1>
                        <h4 data-toc-skip class="author"><h4>
<br>
Authors: <i>Brian M. Schilder, Alan Murphy, Julien Bryois &amp; Nathan
Skene</i><br>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
<br>
Updated: <i>Aug-20-2023</i><br>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MAGMA_Celltyping/blob/HEAD/vignettes/MAGMA.Celltyping.Rmd" class="external-link"><code>vignettes/MAGMA.Celltyping.Rmd</code></a></small>
      <div class="hidden name"><code>MAGMA.Celltyping.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/MAGMA_Celltyping" class="external-link">MAGMA.Celltyping</a></span><span class="op">)</span> </span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p><code>MAGMA.Celltyping</code> is a software package that facilitates
conducting cell-type-specific enrichment tests on GWAS summary
statistics.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Specify where you want the large files to be downloaded to.</p>
<p><em>NOTE</em>: Make sure you change <code>storage_dir</code> to
somewhere other than <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code> if you want to make sure the
results aren’t deleted after this R session closes!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">storage_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-data">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h2>
<div class="section level3">
<h3 id="gwas">GWAS<a class="anchor" aria-label="anchor" href="#gwas"></a>
</h3>
<ul>
<li>We need to have a summary statistics file to analyse as input.<br>
</li>
<li>As an example, you can download UK Biobank summary statistics for
‘fluid_intelligence’ using <code><a href="../reference/get_example_gwas.html">get_example_gwas()</a></code>.</li>
</ul>
<p>Here we provide a pre-munged version of the above file.</p>
<div class="section level4">
<h4 id="munging">Munging<a class="anchor" aria-label="anchor" href="#munging"></a>
</h4>
<p>Our lab have created <a href="https://github.com/neurogenomics/MungeSumstats" class="external-link"><code>MungeSumstats</code></a>,
a robust Bioconductor package for formatting multiple types of summary
statistics files. We highly recommend processing your GWAS summary
statistics with <code>MungeSumstats</code> before continuing. See the
<em>full_workflow</em> vignette for more details.</p>
<p>The minimum info needed <em>after</em> munging is:<br>
- “SNP”, “CHR”, and “BP” as first three columns. - It has at least one
of these columns: “Z”,“OR”,“BETA”,“LOG_ODDS”,“SIGNED_SUMSTAT”</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path_formatted</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_example_gwas.html">get_example_gwas</a></span><span class="op">(</span></span>
<span>  trait <span class="op">=</span> <span class="st">"prospective_memory"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="map-snps-to-genes">Map SNPs to Genes<a class="anchor" aria-label="anchor" href="#map-snps-to-genes"></a>
</h4>
<p>Note you can input the genome build of your summary statistics for
this step or it can be inferred if left <code>NULL</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genesOutPath</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/map_snps_to_genes.html">map_snps_to_genes</a></span><span class="op">(</span></span>
<span>  path_formatted <span class="op">=</span> <span class="va">path_formatted</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="st">"GRCh37"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="magma_files_public">MAGMA_Files_Public<a class="anchor" aria-label="anchor" href="#magma_files_public"></a>
</h4>
<p>Rather than preprocessing the GWAS yourself, you can instead use the
<a href="https://github.com/neurogenomics/MAGMA_Files_Public" class="external-link"><code>MAGMA_Files_Public</code></a>
database we have created. It contains pre-computed MAGMA SNP-to-genes
mapping files for hundreds of GWAS.</p>
<p>You can browse which GWAS traits are available by looking at the
provided <a href="https://github.com/neurogenomics/MAGMA_Files_Public/blob/master/metadata.csv" class="external-link"><em>metadata.csv</em></a>
file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">magma_dirs</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/import_magma_files.html">import_magma_files</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="st">"ieu-a-298"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using built-in example files: ieu-a-298.tsv.gz.35UP.10DOWN</span></span></code></pre>
<pre><code><span><span class="co">## Returning MAGMA directories.</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="celltypedataset">CellTypeDataset<a class="anchor" aria-label="anchor" href="#celltypedataset"></a>
</h3>
<p><code>ewceData</code> provides a number of CellTypeDatasets (CTD) to
be used a cell-type transcriptomic signature reference files.</p>
<p>If you want to create your own single-cell transcriptomic reference,
you’ll need to first convert it to CTD using the instructions found in
the <code>EWCE</code> package <a href="https://github.com/NathanSkene/EWCE/" class="external-link">documentation here</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<p>Note that the cell type dataset loaded in the code above is the
Karolinksa cortex/hippocampus data only. For the full Karolinska dataset
with hypothalamus and midbrain instead use the following:</p>
<pre><code><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_allKI"</span><span class="op">)</span></span></code></pre>
<p>Or for the DRONC seq or AIBS datasets use:</p>
<pre><code><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Tasic"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DivSeq"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_AIBS"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DRONC_human"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_DRONC_mouse"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_BlueLake2018_FrontalCortexOnly"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_BlueLake2018_VisualCortexOnly"</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ctd.html">get_ctd</a></span><span class="op">(</span><span class="st">"ctd_Saunders"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="run-cell-type-enrichment-analyses">Run cell-type enrichment analyses<a class="anchor" aria-label="anchor" href="#run-cell-type-enrichment-analyses"></a>
</h2>
<p><code>MAGMA.Celltyping</code> offers a suite of functions for
conducting various types of cell-type-specific enrichment tests on GWAS
summary statistics.</p>
<p>The <code>celltype_associations_pipeline</code> wraps several
functions that in previous versions of <code>MAGMA.Celltyping</code> had
to be set up and run separately. These include:</p>
<ul>
<li>
<strong>Linear enrichment</strong>:
<code>calculate_celltype_associations(EnrichmentMode = "linear")</code>
internally. Activated when <code>run_linear=TRUE</code>.</li>
<li>
<strong>Top 10% enrichment</strong>: Uses
<code>calculate_celltype_associations(EnrichmentMode = "Top 10%")</code>
internally. Activated when <code>run_top10=TRUE</code>.</li>
<li>
<strong>Conditional enrichment</strong>: Uses
<code>calculate_conditional_celltype_associations</code> internally.
Activated when <code>run_conditional=TRUE</code>.</li>
</ul>
<p>Thus, <code>celltype_associations_pipeline</code> is designed to make
these analyses easier to run.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MAGMA_results</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/celltype_associations_pipeline.html">celltype_associations_pipeline</a></span><span class="op">(</span></span>
<span>  magma_dirs <span class="op">=</span> <span class="va">magma_dirs</span>,</span>
<span>  ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  ctd_species <span class="op">=</span> <span class="st">"mouse"</span>, </span>
<span>  ctd_name <span class="op">=</span> <span class="st">"Zeisel2015"</span>, </span>
<span>  run_linear <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  run_top10 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>We’ve also saved a pre-computed version of these results as a
dataset:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MAGMA_results</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="va"><a href="../reference/enrichment_results.html">enrichment_results</a></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plot-results">Plot results<a class="anchor" aria-label="anchor" href="#plot-results"></a>
</h2>
<div class="section level3">
<h3 id="merge-results">Merge results<a class="anchor" aria-label="anchor" href="#merge-results"></a>
</h3>
<p><code>merge_results</code> imports each of the MAGMA enrichment
results files and merges them into one so that they can easily be
plotted and further analysed.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged_results</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_results.html">merge_results</a></span><span class="op">(</span></span>
<span>  MAGMA_results <span class="op">=</span> <span class="va">MAGMA_results</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Saving full merged results to ==&gt; /tmp/RtmpfhgSUF/MAGMA_celltyping./.lvl1.csv</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">merged_results</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="9%">
<col width="12%">
<col width="1%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="2%">
<col width="3%">
<col width="1%">
<col width="2%">
<col width="4%">
<col width="20%">
<col width="2%">
<col width="4%">
<col width="4%">
<col width="5%">
<col width="2%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">GWAS</th>
<th align="left">Celltype</th>
<th align="left">TYPE</th>
<th align="left">OBS_GENES</th>
<th align="right">BETA</th>
<th align="right">BETA_STD</th>
<th align="right">SE</th>
<th align="right">P</th>
<th align="right">log10p</th>
<th align="right">level</th>
<th align="left">Method</th>
<th align="left">EnrichmentMode</th>
<th align="left">GCOV_FILE</th>
<th align="left">CONTROL</th>
<th align="left">CONTROL_label</th>
<th align="left">genesOutCOND</th>
<th align="left">analysis_name</th>
<th align="right">FDR</th>
<th align="left">Celltype_id</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">0.0016776</td>
<td align="right">0.0199640</td>
<td align="right">0.0024613</td>
<td align="right">0.24785</td>
<td align="right">-0.6058111</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.867475</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">SET</td>
<td align="left">104</td>
<td align="right">0.0671670</td>
<td align="right">0.0178860</td>
<td align="right">0.0853000</td>
<td align="right">0.21562</td>
<td align="right">-0.6663110</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.867475</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">SET</td>
<td align="left">98</td>
<td align="right">0.0831670</td>
<td align="right">0.0215500</td>
<td align="right">0.0962320</td>
<td align="right">0.19384</td>
<td align="right">-0.7125566</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.867475</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">SET</td>
<td align="left">94</td>
<td align="right">0.1124000</td>
<td align="right">0.0285690</td>
<td align="right">0.1017200</td>
<td align="right">0.13472</td>
<td align="right">-0.8705679</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.867475</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0003729</td>
<td align="right">-0.0045268</td>
<td align="right">0.0022566</td>
<td align="right">0.56561</td>
<td align="right">-0.2474829</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">0.0002391</td>
<td align="right">0.0029144</td>
<td align="right">0.0023811</td>
<td align="right">0.46002</td>
<td align="right">-0.3372233</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0033333</td>
<td align="right">-0.0393200</td>
<td align="right">0.0024009</td>
<td align="right">0.91728</td>
<td align="right">-0.0374981</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">interneurons</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0002839</td>
<td align="right">-0.0036131</td>
<td align="right">0.0022446</td>
<td align="right">0.55030</td>
<td align="right">-0.2594005</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">microglia</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0037493</td>
<td align="right">-0.0440450</td>
<td align="right">0.0023802</td>
<td align="right">0.94218</td>
<td align="right">-0.0258661</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0023657</td>
<td align="right">-0.0282810</td>
<td align="right">0.0023306</td>
<td align="right">0.84480</td>
<td align="right">-0.0732461</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">pyramidal_SS</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">SET</td>
<td align="left">106</td>
<td align="right">-0.1554800</td>
<td align="right">-0.0417660</td>
<td align="right">0.0911160</td>
<td align="right">0.95587</td>
<td align="right">-0.0196012</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.981580</td>
<td align="left">interneurons</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">SET</td>
<td align="left">91</td>
<td align="right">-0.0341170</td>
<td align="right">-0.0085425</td>
<td align="right">0.1093000</td>
<td align="right">0.62250</td>
<td align="right">-0.2058606</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.981580</td>
<td align="left">microglia</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">SET</td>
<td align="left">111</td>
<td align="right">-0.1930200</td>
<td align="right">-0.0529550</td>
<td align="right">0.0923320</td>
<td align="right">0.98158</td>
<td align="right">-0.0080743</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.981580</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">SET</td>
<td align="left">98</td>
<td align="right">-0.0750110</td>
<td align="right">-0.0194370</td>
<td align="right">0.0965900</td>
<td align="right">0.78120</td>
<td align="right">-0.1072378</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.981580</td>
<td align="left">pyramidal_SS</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>Now we’ll construct a heatmap visualizing the enrichment results,
such that each GWAS is shown on the y-axis and each cell-type is shown
on the x-axis. Results can be further facetted by what kind of test was
run (linear, top10%, and/or conditional).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">heat</span> <span class="op">&lt;-</span> <span class="fu">MAGMA.Celltyping</span><span class="fu">::</span><span class="fu"><a href="../reference/results_heatmap.html">results_heatmap</a></span><span class="op">(</span></span>
<span>  merged_results <span class="op">=</span> <span class="va">merged_results</span>, </span>
<span>  title <span class="op">=</span> <span class="st">"Alzheimer's Disease (ieu-a-298) vs. nervous system cell-types (Zeisel2015)"</span>,</span>
<span>  fdr_thresh <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 14 results @ FDR &lt; 1</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The `facets` argument of `facet_grid()` is deprecated as of ggplot2 2.2.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use the `rows` argument instead.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">MAGMA.Celltyping</span> package.</span></span>
<span><span class="co">##   Please report the issue at</span></span>
<span><span class="co">##   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/neurogenomics/MAGMA_Celltyping/issues&gt;</span>.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<p><img src="MAGMA.Celltyping_files/figure-html/Heatmap-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="top-results">Top results<a class="anchor" aria-label="anchor" href="#top-results"></a>
</h2>
<div class="section level3">
<h3 id="top-phenotypes">Top phenotypes<a class="anchor" aria-label="anchor" href="#top-phenotypes"></a>
</h3>
<p>Get the phenotypes with the greatest number of significant cell-type
enrichment results.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_phenos</span> <span class="op">&lt;-</span> <span class="va">merged_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">EnrichmentMode</span>, <span class="va">GWAS</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Celltype<span class="op">=</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">Celltype</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">Celltype</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `summarise()` has grouped output by 'EnrichmentMode'. You can override using</span></span>
<span><span class="co">## the `.groups` argument.</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">top_phenos</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">EnrichmentMode</th>
<th align="left">GWAS</th>
<th align="right">Celltype</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="right">7</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="top-enrichments">Top enrichments<a class="anchor" aria-label="anchor" href="#top-enrichments"></a>
</h3>
<p>Get the phenotypes-celltype enrichment results with the most
significant p-values (per phenotype).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_enrich</span> <span class="op">&lt;-</span> <span class="va">merged_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">EnrichmentMode</span>, <span class="va">GWAS</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_min</a></span><span class="op">(</span><span class="va">FDR</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">top_enrich</span><span class="op">)</span> </span></code></pre></div>
<table class="table">
<colgroup>
<col width="9%">
<col width="12%">
<col width="1%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="2%">
<col width="3%">
<col width="1%">
<col width="2%">
<col width="4%">
<col width="20%">
<col width="2%">
<col width="4%">
<col width="4%">
<col width="5%">
<col width="2%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">GWAS</th>
<th align="left">Celltype</th>
<th align="left">TYPE</th>
<th align="left">OBS_GENES</th>
<th align="right">BETA</th>
<th align="right">BETA_STD</th>
<th align="right">SE</th>
<th align="right">P</th>
<th align="right">log10p</th>
<th align="right">level</th>
<th align="left">Method</th>
<th align="left">EnrichmentMode</th>
<th align="left">GCOV_FILE</th>
<th align="left">CONTROL</th>
<th align="left">CONTROL_label</th>
<th align="left">genesOutCOND</th>
<th align="left">analysis_name</th>
<th align="right">FDR</th>
<th align="left">Celltype_id</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">0.0016776</td>
<td align="right">0.0199640</td>
<td align="right">0.0024613</td>
<td align="right">0.24785</td>
<td align="right">-0.6058111</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.867475</td>
<td align="left">oligodendrocytes</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0003729</td>
<td align="right">-0.0045268</td>
<td align="right">0.0022566</td>
<td align="right">0.56561</td>
<td align="right">-0.2474829</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">0.0002391</td>
<td align="right">0.0029144</td>
<td align="right">0.0023811</td>
<td align="right">0.46002</td>
<td align="right">-0.3372233</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">i n t e r n e u r o n s</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0033333</td>
<td align="right">-0.0393200</td>
<td align="right">0.0024009</td>
<td align="right">0.91728</td>
<td align="right">-0.0374981</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">interneurons</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">m i c r o g l i a</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0002839</td>
<td align="right">-0.0036131</td>
<td align="right">0.0022446</td>
<td align="right">0.55030</td>
<td align="right">-0.2594005</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">microglia</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ C A 1</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0037493</td>
<td align="right">-0.0440450</td>
<td align="right">0.0023802</td>
<td align="right">0.94218</td>
<td align="right">-0.0258661</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">pyramidal_CA1</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">p y r a m i d a l _ S S</td>
<td align="left">COVAR</td>
<td align="left">962</td>
<td align="right">-0.0023657</td>
<td align="right">-0.0282810</td>
<td align="right">0.0023306</td>
<td align="right">0.84480</td>
<td align="right">-0.0732461</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Linear</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_linear.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_linear</td>
<td align="right">0.981580</td>
<td align="left">pyramidal_SS</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">a s t r o c y t e s _ e p e n d y m a l</td>
<td align="left">SET</td>
<td align="left">104</td>
<td align="right">0.0671670</td>
<td align="right">0.0178860</td>
<td align="right">0.0853000</td>
<td align="right">0.21562</td>
<td align="right">-0.6663110</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.867475</td>
<td align="left">astrocytes_ependymal</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">e n d o t h e l i a l _ m u r a l</td>
<td align="left">SET</td>
<td align="left">98</td>
<td align="right">0.0831670</td>
<td align="right">0.0215500</td>
<td align="right">0.0962320</td>
<td align="right">0.19384</td>
<td align="right">-0.7125566</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.867475</td>
<td align="left">endothelial_mural</td>
</tr>
<tr class="even">
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN</td>
<td align="left">o l i g o d e n d r o c y t e s</td>
<td align="left">SET</td>
<td align="left">94</td>
<td align="right">0.1124000</td>
<td align="right">0.0285690</td>
<td align="right">0.1017200</td>
<td align="right">0.13472</td>
<td align="right">-0.8705679</td>
<td align="right">1</td>
<td align="left">MAGMA</td>
<td align="left">Top 10%</td>
<td align="left">ieu-a-298.tsv.gz.35UP.10DOWN.level1.Zeisel2015_top10.gsa.out</td>
<td align="left">BASELINE</td>
<td align="left">BASELINE</td>
<td align="left">NA</td>
<td align="left">Zeisel2015_top10</td>
<td align="right">0.867475</td>
<td align="left">oligodendrocytes</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ewceData_1.9.0          ExperimentHub_2.9.1     AnnotationHub_3.9.1    </span></span>
<span><span class="co">## [4] BiocFileCache_2.9.1     dbplyr_2.3.3            BiocGenerics_0.47.0    </span></span>
<span><span class="co">## [7] dplyr_1.1.2             MAGMA.Celltyping_2.0.11 BiocStyle_2.29.1       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.3.1                 later_1.3.1                  </span></span>
<span><span class="co">##   [3] BiocIO_1.11.0                 bitops_1.0-7                 </span></span>
<span><span class="co">##   [5] ggplotify_0.1.2               filelock_1.0.2               </span></span>
<span><span class="co">##   [7] tibble_3.2.1                  R.oo_1.25.0                  </span></span>
<span><span class="co">##   [9] XML_3.99-0.14                 lifecycle_1.0.3              </span></span>
<span><span class="co">##  [11] rstatix_0.7.2                 rprojroot_2.0.3              </span></span>
<span><span class="co">##  [13] MASS_7.3-60                   lattice_0.21-8               </span></span>
<span><span class="co">##  [15] backports_1.4.1               magrittr_2.0.3               </span></span>
<span><span class="co">##  [17] limma_3.57.7                  plotly_4.10.2                </span></span>
<span><span class="co">##  [19] sass_0.4.7                    rmarkdown_2.24               </span></span>
<span><span class="co">##  [21] jquerylib_0.1.4               yaml_2.3.7                   </span></span>
<span><span class="co">##  [23] httpuv_1.6.11                 HGNChelper_0.8.1             </span></span>
<span><span class="co">##  [25] minqa_1.2.5                   DBI_1.1.3                    </span></span>
<span><span class="co">##  [27] abind_1.4-5                   zlibbioc_1.47.0              </span></span>
<span><span class="co">##  [29] GenomicRanges_1.53.1          purrr_1.0.2                  </span></span>
<span><span class="co">##  [31] R.utils_2.12.2                RCurl_1.98-1.12              </span></span>
<span><span class="co">##  [33] yulab.utils_0.0.7             VariantAnnotation_1.47.1     </span></span>
<span><span class="co">##  [35] rappdirs_0.3.3                GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">##  [37] IRanges_2.35.2                S4Vectors_0.39.1             </span></span>
<span><span class="co">##  [39] tidytree_0.4.5                pkgdown_2.0.7                </span></span>
<span><span class="co">##  [41] codetools_0.2-19              DelayedArray_0.27.10         </span></span>
<span><span class="co">##  [43] xml2_1.3.5                    tidyselect_1.2.0             </span></span>
<span><span class="co">##  [45] aplot_0.2.0                   farver_2.1.1                 </span></span>
<span><span class="co">##  [47] lme4_1.1-34                   matrixStats_1.0.0            </span></span>
<span><span class="co">##  [49] stats4_4.3.1                  GenomicAlignments_1.37.0     </span></span>
<span><span class="co">##  [51] jsonlite_1.8.7                ellipsis_0.3.2               </span></span>
<span><span class="co">##  [53] systemfonts_1.0.4             tools_4.3.1                  </span></span>
<span><span class="co">##  [55] progress_1.2.2                treeio_1.25.3                </span></span>
<span><span class="co">##  [57] ragg_1.2.5                    Rcpp_1.0.11                  </span></span>
<span><span class="co">##  [59] glue_1.6.2                    SparseArray_1.1.11           </span></span>
<span><span class="co">##  [61] xfun_0.40                     MatrixGenerics_1.13.1        </span></span>
<span><span class="co">##  [63] GenomeInfoDb_1.37.2           RNOmni_1.0.1                 </span></span>
<span><span class="co">##  [65] withr_2.5.0                   BiocManager_1.30.22          </span></span>
<span><span class="co">##  [67] fastmap_1.1.1                 boot_1.3-28.1                </span></span>
<span><span class="co">##  [69] fansi_1.0.4                   digest_0.6.33                </span></span>
<span><span class="co">##  [71] R6_2.5.1                      mime_0.12                    </span></span>
<span><span class="co">##  [73] gridGraphics_0.5-1            textshaping_0.3.6            </span></span>
<span><span class="co">##  [75] colorspace_2.1-0              biomaRt_2.57.1               </span></span>
<span><span class="co">##  [77] RSQLite_2.3.1                 R.methodsS3_1.8.2            </span></span>
<span><span class="co">##  [79] utf8_1.2.3                    tidyr_1.3.0                  </span></span>
<span><span class="co">##  [81] generics_0.1.3                data.table_1.14.8            </span></span>
<span><span class="co">##  [83] rtracklayer_1.61.1            prettyunits_1.1.1            </span></span>
<span><span class="co">##  [85] httr_1.4.7                    htmlwidgets_1.6.2            </span></span>
<span><span class="co">##  [87] S4Arrays_1.1.5                pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [89] gtable_0.3.3                  blob_1.2.4                   </span></span>
<span><span class="co">##  [91] SingleCellExperiment_1.23.0   XVector_0.41.1               </span></span>
<span><span class="co">##  [93] htmltools_0.5.6               carData_3.0-5                </span></span>
<span><span class="co">##  [95] bookdown_0.35                 scales_1.2.1                 </span></span>
<span><span class="co">##  [97] Biobase_2.61.0                png_0.1-8                    </span></span>
<span><span class="co">##  [99] ggdendro_0.1.23               ggfun_0.1.2                  </span></span>
<span><span class="co">## [101] knitr_1.43                    reshape2_1.4.4               </span></span>
<span><span class="co">## [103] rjson_0.2.21                  nloptr_2.0.3                 </span></span>
<span><span class="co">## [105] nlme_3.1-163                  curl_5.0.2                   </span></span>
<span><span class="co">## [107] cachem_1.0.8                  stringr_1.5.0                </span></span>
<span><span class="co">## [109] BiocVersion_3.18.0            parallel_4.3.1               </span></span>
<span><span class="co">## [111] AnnotationDbi_1.63.2          restfulr_0.0.15              </span></span>
<span><span class="co">## [113] desc_1.4.2                    pillar_1.9.0                 </span></span>
<span><span class="co">## [115] grid_4.3.1                    vctrs_0.6.3                  </span></span>
<span><span class="co">## [117] promises_1.2.1                ggpubr_0.6.0                 </span></span>
<span><span class="co">## [119] car_3.1-2                     xtable_1.8-4                 </span></span>
<span><span class="co">## [121] evaluate_0.21                 orthogene_1.7.0              </span></span>
<span><span class="co">## [123] GenomicFeatures_1.53.1        cli_3.6.1                    </span></span>
<span><span class="co">## [125] compiler_4.3.1                Rsamtools_2.17.0             </span></span>
<span><span class="co">## [127] rlang_1.1.1                   crayon_1.5.2                 </span></span>
<span><span class="co">## [129] grr_0.9.5                     ggsignif_0.6.4               </span></span>
<span><span class="co">## [131] labeling_0.4.2                gprofiler2_0.2.2             </span></span>
<span><span class="co">## [133] EWCE_1.9.2                    plyr_1.8.8                   </span></span>
<span><span class="co">## [135] fs_1.6.3                      stringi_1.7.12               </span></span>
<span><span class="co">## [137] viridisLite_0.4.2             BiocParallel_1.35.4          </span></span>
<span><span class="co">## [139] assertthat_0.2.1              babelgene_22.9               </span></span>
<span><span class="co">## [141] munsell_0.5.0                 Biostrings_2.69.2            </span></span>
<span><span class="co">## [143] gh_1.4.0                      lazyeval_0.2.2               </span></span>
<span><span class="co">## [145] homologene_1.4.68.19.3.27     Matrix_1.6-1                 </span></span>
<span><span class="co">## [147] MungeSumstats_1.9.15          BSgenome_1.69.0              </span></span>
<span><span class="co">## [149] hms_1.1.3                     patchwork_1.1.3              </span></span>
<span><span class="co">## [151] bit64_4.0.5                   ggplot2_3.4.3                </span></span>
<span><span class="co">## [153] KEGGREST_1.41.0               statmod_1.5.0                </span></span>
<span><span class="co">## [155] shiny_1.7.5                   highr_0.10                   </span></span>
<span><span class="co">## [157] SummarizedExperiment_1.31.1   interactiveDisplayBase_1.39.0</span></span>
<span><span class="co">## [159] googleAuthR_2.0.1             gargle_1.5.2                 </span></span>
<span><span class="co">## [161] broom_1.0.5                   memoise_2.0.1                </span></span>
<span><span class="co">## [163] bslib_0.5.1                   ggtree_3.9.1                 </span></span>
<span><span class="co">## [165] bit_4.0.5                     ape_5.7-1</span></span></code></pre>
</details><hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Julien Bryois, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
